{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "9d47b8d8-edf0-4b16-a5a3-3d37ca6e9c91",
   "metadata": {},
   "outputs": [],
   "source": [
    "#!/usr/bin/env python3\n",
    "'''Match binding sites to scaffolds\n",
    "using standard Rosetta matcher.\n",
    "'''\n",
    "import os\n",
    "import subprocess\n",
    "import shutil\n",
    "import json\n",
    "\n",
    "import numpy as np\n",
    "\n",
    "import pyrosetta\n",
    "from pyrosetta import rosetta\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "48e3b29b-f4ee-47c9-87f8-15c268200353",
   "metadata": {},
   "outputs": [],
   "source": [
    "def angle(x1, x2, x3):\n",
    "    '''Return the angle x1-x2-x3 in degrees.\n",
    "    x1, x2 and x3 are xyzVectors.\n",
    "    '''\n",
    "    v1 = (x1 - x2).normalized()\n",
    "    v2 = (x3 - x2).normalized()\n",
    "\n",
    "    return np.arccos(v1.dot(v2)) * 180 / np.pi\n",
    "\n",
    "def dihedral(x1, x2, x3, x4):\n",
    "    '''Return the dihedral x1-x2-x3-x4 in degrees.\n",
    "    x1, x2, x3 and x4 are xyzVectors.\n",
    "    '''\n",
    "    v1 = (x1 - x2).normalized()\n",
    "    v2 = (x3 - x2).normalized()\n",
    "    v3 = (x4 - x3).normalized()\n",
    "\n",
    "    v1_parallel = v2.normalized()\n",
    "    v1_parallel *= v1.dot(v2)\n",
    "    v3_parallel = v2.normalized()\n",
    "    v3_parallel *= v3.dot(v2)\n",
    "    \n",
    "    v1_n = (v1 - v1_parallel).normalized()\n",
    "    v3_n = (v3 - v3_parallel).normalized()\n",
    "\n",
    "    cos = v1_n.dot(v3_n)\n",
    "    sin = v1_n.cross(v3_n).dot(v2)\n",
    "\n",
    "    return np.arctan2(sin, cos) * 180 / np.pi\n",
    "\n",
    "\n",
    "def atom_residue_distance(pose, res, atom_id, res2):\n",
    "    '''Return the minimum heavy atom distance\n",
    "    between an atom and a residue.\n",
    "    '''\n",
    "    min_distance = float('inf')\n",
    "\n",
    "    for a2 in range(1, pose.residue(res2).nheavyatoms() + 1):\n",
    "        dist = pose.residue(res).xyz(atom_id).distance(pose.residue(res2).xyz(a2))\n",
    "\n",
    "        if dist < min_distance:\n",
    "            min_distance = dist\n",
    "\n",
    "    return min_distance\n",
    "\n",
    "def find_contact_heavy_atoms(pose, res1, res2):\n",
    "    '''Find 3 contact heavy atoms for a residue \n",
    "     i.e. the heavy atoms that is closest to the opposing residue\n",
    "     and the 2 heavy atoms that are closest to \n",
    "     this residue\n",
    "    '''\n",
    "    # Get the atoms that are can be considered as contact atoms for residue 1\n",
    "\n",
    "    potential_contact_atom_ids = []\n",
    "\n",
    "    for a in range(1, pose.residue(res1).nheavyatoms() + 1):\n",
    "        # Exclude protein backbone atoms except for N, CA and C\n",
    "        # If backbon variants are included, for example OXT, the\n",
    "        # matcher application would crash\n",
    "        \n",
    "        if pose.residue(res1).is_protein():\n",
    "            if pose.residue(res1).atom_is_backbone(a):\n",
    "                if pose.residue(res1).atom_name(a).strip() in ['N', 'CA', 'C']:\n",
    "                    potential_contact_atom_ids.append(a)\n",
    "            elif not pose.residue(res1).is_virtual(a): # Ignore virtual atoms for protein residues\n",
    "                potential_contact_atom_ids.append(a)\n",
    "\n",
    "        else:\n",
    "            potential_contact_atom_ids.append(a)\n",
    "\n",
    "    # Calculate distances for all heavy atoms\n",
    "    \n",
    "    distances = []\n",
    "\n",
    "    for a in potential_contact_atom_ids:\n",
    "        distances.append((a, atom_residue_distance(pose, res1, a, res2)))\n",
    "\n",
    "    # Get the names of the closest\n",
    "\n",
    "    distances_sorted = sorted(distances, key=lambda x : x[1])\n",
    "\n",
    "    contact_atoms = []\n",
    "    contact_atoms.append(pose.residue(res1).atom_name(distances_sorted[0][0]))\n",
    "\n",
    "    # Get the two closest intra-residue atoms\n",
    "\n",
    "    intra_distances = []\n",
    "\n",
    "    for a in potential_contact_atom_ids:\n",
    "        if a == distances_sorted[0][0]:\n",
    "            continue\n",
    "\n",
    "        intra_distances.append((a, pose.residue(res1).xyz(distances_sorted[0][0]).distance(\n",
    "            pose.residue(res1).xyz(a))))\n",
    "\n",
    "    intra_distances_sorted = sorted(intra_distances, key=lambda x:x[1])\n",
    "\n",
    "    for i in range(2):\n",
    "        contact_atoms.append(pose.residue(res1).atom_name(intra_distances_sorted[i][0]))\n",
    "\n",
    "    return contact_atoms\n",
    "\n",
    "def generate_cst_file(site_pose, output_file):\n",
    "    '''Generate the constraint file for Rosetta matching.'''\n",
    "    # Find the ligand and protein residues\n",
    "\n",
    "    ligand_residue = 0\n",
    "    protein_residues = []\n",
    "    for i in range(1, site_pose.size() + 1):\n",
    "        if site_pose.residue(i).is_ligand():\n",
    "            ligand_residue = i\n",
    "        elif site_pose.residue(i).is_protein():\n",
    "            protein_residues.append(i)\n",
    "\n",
    "    assert(ligand_residue != 0)\n",
    "\n",
    "    # Generate constraints between the ligand and binding site protein residues \n",
    "\n",
    "    template_sections = \\\n",
    "'''CST::BEGIN\n",
    "  TEMPLATE::   ATOM_MAP: 1 atom_name: {a1_1} {a1_2} {a1_3}\n",
    "  TEMPLATE::   ATOM_MAP: 1 residue3: {r1}\n",
    "\n",
    "  TEMPLATE::   ATOM_MAP: 2 atom_name: {a2_1} {a2_2} {a2_3}\n",
    "  TEMPLATE::   ATOM_MAP: 2 residue3: {r2}\n",
    "\n",
    "  CONSTRAINT:: distanceAB: {dist:7.2f}   0.30 100.00  0        0\n",
    "  CONSTRAINT::    angle_A: {angA:7.2f}  15.00 100.00  360.00   1\n",
    "  CONSTRAINT::    angle_B: {angB:7.2f}  15.00 100.00  360.00   1\n",
    "  CONSTRAINT::  torsion_A: {torA:7.2f}  15.00 100.00  360.00   1\n",
    "  CONSTRAINT::  torsion_B: {torB:7.2f}  15.00 100.00  360.00   1\n",
    "  CONSTRAINT:: torsion_AB: {toAB:7.2f}  15.00 100.00  360.00   1\n",
    "CST::END'''\n",
    "\n",
    "    constraint_sections = []\n",
    "\n",
    "    for pro_res in protein_residues:\n",
    "        ligand_contact_atoms = find_contact_heavy_atoms(site_pose, ligand_residue, pro_res)\n",
    "        protein_contact_atoms = find_contact_heavy_atoms(site_pose, pro_res, ligand_residue)\n",
    "\n",
    "        # Get the positions of contact atoms\n",
    "\n",
    "        v_r1a1 = site_pose.residue(ligand_residue).xyz(ligand_contact_atoms[0])\n",
    "        v_r1a2 = site_pose.residue(ligand_residue).xyz(ligand_contact_atoms[1])\n",
    "        v_r1a3 = site_pose.residue(ligand_residue).xyz(ligand_contact_atoms[2])\n",
    "\n",
    "        v_r2a1 = site_pose.residue(pro_res).xyz(protein_contact_atoms[0])\n",
    "        v_r2a2 = site_pose.residue(pro_res).xyz(protein_contact_atoms[1])\n",
    "        v_r2a3 = site_pose.residue(pro_res).xyz(protein_contact_atoms[2])\n",
    "\n",
    "        # Calculate the constraint distance, angles and torsions\n",
    "\n",
    "        dAB = v_r1a1.distance(v_r2a1)\n",
    "        aA = angle(v_r1a2, v_r1a1, v_r2a1)\n",
    "        aB = angle(v_r1a1, v_r2a1, v_r2a2)\n",
    "        tA = dihedral(v_r1a3, v_r1a2, v_r1a1, v_r2a1)\n",
    "        tB = dihedral(v_r1a1, v_r2a1, v_r2a2, v_r2a3)\n",
    "        tAB = dihedral(v_r1a2, v_r1a1, v_r2a1, v_r2a2)\n",
    "\n",
    "        # Generate the constraint file section\n",
    "\n",
    "        constraint_section = template_sections.format(\n",
    "                a1_1=ligand_contact_atoms[0], a1_2=ligand_contact_atoms[1], a1_3=ligand_contact_atoms[2], r1=site_pose.residue(ligand_residue).name3(),\n",
    "                a2_1=protein_contact_atoms[0], a2_2=protein_contact_atoms[1], a2_3=protein_contact_atoms[2], r2=site_pose.residue(pro_res).name3(),\n",
    "                dist=dAB, angA=aA, angB=aB, torA=tA, torB=tB, toAB=tAB)\n",
    "\n",
    "        constraint_sections.append(constraint_section)\n",
    "\n",
    "    # Write the constraint file\n",
    "\n",
    "    with open(output_file, 'w') as f:\n",
    "        f.write('\\n\\n'.join(constraint_sections))\n",
    "\n",
    "def write_params_file_for_ligand(binding_site_pdb, output_file):\n",
    "    '''Write a params file for the ligand residue in the binding site.\n",
    "    Carefull!!! The default output is not aligned and does not work for matcher.\n",
    "    '''\n",
    "    site_pose = pyrosetta.pose_from_file(binding_site_pdb) \n",
    "    \n",
    "    # Find the ligand and protein residues\n",
    "\n",
    "    ligand_residue = 0\n",
    "    for i in range(1, site_pose.size() + 1):\n",
    "        if site_pose.residue(i).is_ligand():\n",
    "            ligand_residue = i\n",
    "\n",
    "    assert(ligand_residue != 0)\n",
    "\n",
    "    # Write the .params file\n",
    "\n",
    "    rosetta.core.chemical.write_topology_file(site_pose.residue(ligand_residue).type(), output_file)\n",
    "\n",
    "def standard_rosetta_match(scratching_path, scaffold_pdb, constraint_file, ligand_name, params_file=None, keep_outputs=False):\n",
    "    '''Run standard Rosetta matching.\n",
    "    Return the number of matches and the matched positions in rosetta numbering.\n",
    "    Return -1 if matching failed.\n",
    "    '''\n",
    "    abs_scaffold_pdb = os.path.abspath(scaffold_pdb)\n",
    "    abs_constraint_file = os.path.abspath(constraint_file)\n",
    "    \n",
    "    if not (params_file is None):\n",
    "        abs_params_file = os.path.abspath(params_file)\n",
    "   \n",
    "    with open('./site_settings.json', 'r') as f:\n",
    "        site_settings = json.load(f)\n",
    "\n",
    "        matcher_app = site_settings['matcher_app']\n",
    "\n",
    "    # Go to the output path\n",
    "\n",
    "    cwd = os.getcwd()\n",
    "\n",
    "    os.makedirs(scratching_path, exist_ok=True)\n",
    "    os.chdir(scratching_path)\n",
    "\n",
    "    # Generate pos file\n",
    "\n",
    "    scaffold_pose = pyrosetta.pose_from_file(abs_scaffold_pdb)\n",
    "    scaffold_protein_residues = []\n",
    "\n",
    "    for i in range(1, scaffold_pose.size() + 1):\n",
    "        if scaffold_pose.residue(i).is_protein():\n",
    "            scaffold_protein_residues.append(str(i))\n",
    "\n",
    "    pos_file = 'scaffold.pos'\n",
    "    with open(pos_file, 'w') as f:\n",
    "        f.write(' '.join(scaffold_protein_residues))\n",
    "\n",
    "    # Run matcher\n",
    "        \n",
    "    matcher_cmd = [matcher_app,\n",
    "        '-match:output_format', 'PDB',\n",
    "        '-match:match_grouper', 'SameSequenceGrouper',\n",
    "        '-match:consolidate_matches',\n",
    "        '-match:output_matches_per_group', '1',\n",
    "        '-use_input_sc',\n",
    "        '-in:ignore_unrecognized_res',\n",
    "        '-ex1', '-ex2',\n",
    "        '-enumerate_ligand_rotamers', 'false',\n",
    "        '-match::lig_name', ligand_name,\n",
    "        '-match:geometric_constraint_file', abs_constraint_file,\n",
    "        '-s', abs_scaffold_pdb,\n",
    "        '-match::scaffold_active_site_residues', pos_file\n",
    "        ]\n",
    "\n",
    "    if not (params_file is None):\n",
    "        matcher_cmd += ['-extra_res_fa', abs_params_file]\n",
    "\n",
    "    return_code = subprocess.call(matcher_cmd)\n",
    "\n",
    "    # Count the number of matches and get the matched positions\n",
    "\n",
    "    num_matches = 0\n",
    "    matched_positions = []\n",
    "    scaffold_pose = pyrosetta.pose_from_file(abs_scaffold_pdb)\n",
    "    scaffold_pose.pdb_info().set_chains('A')\n",
    "\n",
    "    for f in os.listdir('.'):\n",
    "        if f.startswith('UM') and f.endswith('.pdb'):\n",
    "            num_matches += 1\n",
    "\n",
    "            # Extract the matched positions from the name of the output file\n",
    "\n",
    "            matched_positions.append([])\n",
    "            pos_string = f.split('_')[2]\n",
    "           \n",
    "            # Get the positions of the amino acid characters\n",
    "            c_positions = []\n",
    "            for i in range(len(pos_string)):\n",
    "                if not pos_string[i].isnumeric():\n",
    "                    c_positions.append(i)\n",
    "\n",
    "            for i in range(len(c_positions)):\n",
    "                if i < len(c_positions) - 1:\n",
    "                    pdb_id = int(pos_string[c_positions[i] + 1 : c_positions[i + 1]])\n",
    "                else:\n",
    "                    pdb_id = int(pos_string[c_positions[i] + 1 : len(pos_string)])\n",
    "\n",
    "                matched_positions[-1].append(scaffold_pose.pdb_info().pdb2pose('A', pdb_id))\n",
    "\n",
    "    # Clear the scratch path\n",
    "\n",
    "    os.chdir(cwd)\n",
    "\n",
    "    if not keep_outputs:\n",
    "        shutil.rmtree(scratching_path)\n",
    "\n",
    "    if 0 == return_code:\n",
    "        return num_matches, matched_positions\n",
    "\n",
    "    else:\n",
    "        return -1, matched_positions\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c1cd6dce-05d5-44ea-b22f-2c965b0589c8",
   "metadata": {},
   "source": [
    "### main"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "a3edc994-5c47-44c2-90f7-4d0c88523da6",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2021 [Rosetta PyRosetta4.conda.mac.cxx11thread.serialization.python39.Release 2021.38+release.4d5a9693c031951766d575fca9286ed34f560779 2021-09-22T20:45:44] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "core.init: {0} Checking for fconfig files in pwd and ./rosetta/flags\n",
      "core.init: {0} Rosetta version: PyRosetta4.conda.mac.cxx11thread.serialization.python39.Release r297 2021.38+release.4d5a9693c03 4d5a9693c031951766d575fca9286ed34f560779 http://www.pyrosetta.org 2021-09-22T20:45:44\n",
      "core.init: {0} command: PyRosetta -ex1 -ex2aro -database /Users/chenlin/opt/anaconda3/envs/MD/lib/python3.9/site-packages/pyrosetta/database\n",
      "basic.random.init_random_generator: {0} 'RNG device' seed mode, using '/dev/urandom', seed=810799639 seed_offset=0 real_seed=810799639 thread_index=0\n",
      "basic.random.init_random_generator: {0} RandomGenerator:init: Normal mode, seed=810799639 RG_type=mt19937\n"
     ]
    }
   ],
   "source": [
    "pyrosetta.init()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "e5ac72d7-5390-4a2c-bbc5-de66a53f5491",
   "metadata": {},
   "outputs": [],
   "source": [
    "binding_site_pdb = './test_inputs/6qga_A_binding_site_bk.pdb'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "4c29ded5-c873-42c2-a650-bf078fd70485",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.chemical.GlobalResidueTypeSet: {0} Finished initializing fa_standard residue type set.  Created 983 residue types\n",
      "core.chemical.GlobalResidueTypeSet: {0} Total time to initialize 0.701426 seconds.\n",
      "core.import_pose.import_pose: {0} File './test_inputs/6qga_A_binding_site_bk.pdb' automatically determined to be of type PDB\n",
      "core.chemical.GlobalResidueTypeSet: {0} Loading (but possibly not actually using) 'J1K' from the PDB components dictionary for residue type 'pdb_J1K'\n",
      "core.io.pose_from_sfr.PoseFromSFRBuilder: {0} Adding undetected upper terminus type to residue 3,  226 A\n",
      "core.conformation.Conformation: {0} [ WARNING ] missing heavyatom:  OXT on residue GLU:CtermProteinFull 3\n"
     ]
    }
   ],
   "source": [
    "site_pose = pyrosetta.pose_from_file(binding_site_pdb) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "0943afd5-f000-4ed3-ad62-6c53cf7dfea6",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.import_pose.import_pose: {0} File './test_inputs/6qga_A_binding_site_bk.pdb' automatically determined to be of type PDB\n",
      "core.io.pose_from_sfr.PoseFromSFRBuilder: {0} Adding undetected upper terminus type to residue 3,  226 A\n",
      "core.conformation.Conformation: {0} [ WARNING ] missing heavyatom:  OXT on residue GLU:CtermProteinFull 3\n"
     ]
    }
   ],
   "source": [
    "write_params_file_for_ligand(binding_site_pdb, './test_inputs/J1K1.params')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "a9316071-5455-438b-9275-ca7df9a64fcb",
   "metadata": {},
   "outputs": [],
   "source": [
    "generate_cst_file(site_pose, './test_inputs/1.cst')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "29860fdb-4bca-4287-a849-3f7bd7ea54da",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "a3e540ee-9ef7-40c8-9e26-86791888f6df",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.import_pose.import_pose: {0} File '/Users/chenlin/OneDrive/thuresearch_git/Match/test_inputs/6qga_B.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: {0} Found disulfide between residues 9 47\n",
      "core.conformation.Conformation: {0} current variant for 9 CYS\n",
      "core.conformation.Conformation: {0} current variant for 47 CYS\n",
      "core.conformation.Conformation: {0} current variant for 9 CYD\n",
      "core.conformation.Conformation: {0} current variant for 47 CYD\n",
      "core.conformation.Conformation: {0} Found disulfide between residues 179 484\n",
      "core.conformation.Conformation: {0} current variant for 179 CYS\n",
      "core.conformation.Conformation: {0} current variant for 484 CYS\n",
      "core.conformation.Conformation: {0} current variant for 179 CYD\n",
      "core.conformation.Conformation: {0} current variant for 484 CYD\n",
      "core.conformation.Conformation: {0} Found disulfide between residues 258 275\n",
      "core.conformation.Conformation: {0} current variant for 258 CYS\n",
      "core.conformation.Conformation: {0} current variant for 275 CYS\n",
      "core.conformation.Conformation: {0} current variant for 258 CYD\n",
      "core.conformation.Conformation: {0} current variant for 275 CYD\n",
      "core.conformation.Conformation: {0} Found disulfide between residues 295 303\n",
      "core.conformation.Conformation: {0} current variant for 295 CYS\n",
      "core.conformation.Conformation: {0} current variant for 303 CYS\n",
      "core.conformation.Conformation: {0} current variant for 295 CYD\n",
      "core.conformation.Conformation: {0} current variant for 303 CYD\n",
      "core.conformation.Conformation: {0} Found disulfide between residues 532 554\n",
      "core.conformation.Conformation: {0} current variant for 532 CYS\n",
      "core.conformation.Conformation: {0} current variant for 554 CYS\n",
      "core.conformation.Conformation: {0} current variant for 532 CYD\n",
      "core.conformation.Conformation: {0} current variant for 554 CYD\n",
      "core.init: (0) Checking for fconfig files in pwd and ./rosetta/flags \n",
      "core.init: (0) Rosetta version: rosetta.source.release-247 r247 2020.08+release.cb1caba cb1cabafd7463ab703f6abf5efa33d2707b85924 https://www.rosettacommons.org 2020-02-22T18:49:33.865989\n",
      "core.init: (0) command: /usr/local/rosetta_mpi/main/source/bin/match.mpi.macosclangrelease -match:output_format PDB -match:match_grouper SameSequenceGrouper -match:consolidate_matches -match:output_matches_per_group 1 -use_input_sc -in:ignore_unrecognized_res -ex1 -ex2 -enumerate_ligand_rotamers false -match::lig_name pdb_J1K -match:geometric_constraint_file /Users/chenlin/OneDrive/thuresearch_git/Match/test_inputs/1.cst -s /Users/chenlin/OneDrive/thuresearch_git/Match/test_inputs/6qga_B.pdb -match::scaffold_active_site_residues scaffold.pos -extra_res_fa /Users/chenlin/OneDrive/thuresearch_git/Match/test_inputs/J1K1.params\n",
      "basic.random.init_random_generator: (0) 'RNG device' seed mode, using '/dev/urandom', seed=359582535 seed_offset=0 real_seed=359582535\n",
      "basic.random.init_random_generator: (0) RandomGenerator:init: Normal mode, seed=359582535 RG_type=mt19937\n",
      "core.init: (0) found database environment variable ROSETTA3_DB: /usr/local/rosetta_mpi/main/database\n",
      "core.chemical: (0) [ WARNING ] Ignoring line starting with 'VARIANT_OF' when parsing topology file.\n",
      "core.chemical.GlobalResidueTypeSet: (0) Finished initializing fa_standard residue type set.  Created 981 residue types\n",
      "core.chemical.GlobalResidueTypeSet: (0) Total time to initialize 0.673369 seconds.\n",
      "core.import_pose.import_pose: (0) File '/Users/chenlin/OneDrive/thuresearch_git/Match/test_inputs/6qga_B.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: (0) Found disulfide between residues 9 47\n",
      "core.conformation.Conformation: (0) current variant for 9 CYS\n",
      "core.conformation.Conformation: (0) current variant for 47 CYS\n",
      "core.conformation.Conformation: (0) current variant for 9 CYD\n",
      "core.conformation.Conformation: (0) current variant for 47 CYD\n",
      "core.conformation.Conformation: (0) Found disulfide between residues 179 484\n",
      "core.conformation.Conformation: (0) current variant for 179 CYS\n",
      "core.conformation.Conformation: (0) current variant for 484 CYS\n",
      "core.conformation.Conformation: (0) current variant for 179 CYD\n",
      "core.conformation.Conformation: (0) current variant for 484 CYD\n",
      "core.conformation.Conformation: (0) Found disulfide between residues 258 275\n",
      "core.conformation.Conformation: (0) current variant for 258 CYS\n",
      "core.conformation.Conformation: (0) current variant for 275 CYS\n",
      "core.conformation.Conformation: (0) current variant for 258 CYD\n",
      "core.conformation.Conformation: (0) current variant for 275 CYD\n",
      "core.conformation.Conformation: (0) Found disulfide between residues 295 303\n",
      "core.conformation.Conformation: (0) current variant for 295 CYS\n",
      "core.conformation.Conformation: (0) current variant for 303 CYS\n",
      "core.conformation.Conformation: (0) current variant for 295 CYD\n",
      "core.conformation.Conformation: (0) current variant for 303 CYD\n",
      "core.conformation.Conformation: (0) Found disulfide between residues 532 554\n",
      "core.conformation.Conformation: (0) current variant for 532 CYS\n",
      "core.conformation.Conformation: (0) current variant for 554 CYS\n",
      "core.conformation.Conformation: (0) current variant for 532 CYD\n",
      "core.conformation.Conformation: (0) current variant for 554 CYD\n",
      "apps.public.match.match: (0) Selecting orientation atoms: CB   C4   O2B \n",
      "protocols.match.MatcherTask: (0) [ WARNING ] no grid file specified for option -grid_boundary. The bounding grid will be generated from the dimensions of the pose. This is experimental at the moment.\n",
      "protocols.match.MatcherTask: (0) Bounding box lower corner set to (-50.0712,-61.0956,-30.449), upper corner set to to (14.845,-6.074,41.4867).\n",
      "protocols.toolbox.match_enzdes_util.EnzConstraintIO: (0) read enzyme constraints from /Users/chenlin/OneDrive/thuresearch_git/Match/test_inputs/1.cst ... done, 3 cst blocks were read.\n",
      "protocols.toolbox.match_enzdes_util.AllowedSeqposForGeomCst: (0) Reading match::scaffold_active_stie_residues scaffold.pos: 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40 41 42 43 44 45 46 47 48 49 50 51 52 53 54 55 56 57 58 59 60 61 62 63 64 65 66 67 68 69 70 71 72 73 74 75 76 77 78 79 80 81 82 83 84 85 86 87 88 89 90 91 92 93 94 95 96 97 98 99 100 101 102 103 104 105 106 107 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251 252 253 254 255 256 257 258 259 260 261 262 263 264 265 266 267 268 269 270 271 272 273 274 275 276 277 278 279 280 281 282 283 284 285 286 287 288 289 290 291 292 293 294 295 296 297 298 299 300 301 302 303 304 305 306 307 308 309 310 311 312 313 314 315 316 317 318 319 320 321 322 323 324 325 326 327 328 329 330 331 332 333 334 335 336 337 338 339 340 341 342 343 344 345 346 347 348 349 350 351 352 353 354 355 356 357 358 359 360 361 362 363 364 365 366 367 368 369 370 371 372 373 374 375 376 377 378 379 380 381 382 383 384 385 386 387 388 389 390 391 392 393 394 395 396 397 398 399 400 401 402 403 404 405 406 407 408 409 410 411 412 413 414 415 416 417 418 419 420 421 422 423 424 425 426 427 428 429 430 431 432 433 434 435 436 437 438 439 440 441 442 443 444 445 446 447 448 449 450 451 452 453 454 455 456 457 458 459 460 461 462 463 464 465 466 467 468 469 470 471 472 473 474 475 476 477 478 479 480 481 482 483 484 485 486 487 488 489 490 491 492 493 494 495 496 497 498 499 500 501 502 503 504 505 506 507 508 509 510 511 512 513 514 515 516 517 518 519 520 521 522 523 524 525 526 527 528 529 530 531 532 533 534 535 536 537 538 539 540 541 542 543 544 545 546 547 548 549 550 551 552 553 554 555 556 557 558\n",
      "protocols.BumpGrid: (0) Atom type CNH2 with ProbeRadius 5 = 1.65\n",
      "protocols.BumpGrid: (0) Atom type COO with ProbeRadius 5 = 1.65\n",
      "protocols.BumpGrid: (0) Atom type CH0 with ProbeRadius 6 = 1.75\n",
      "protocols.BumpGrid: (0) Atom type CH1 with ProbeRadius 6 = 1.75\n",
      "protocols.BumpGrid: (0) Atom type CH2 with ProbeRadius 6 = 1.75\n",
      "protocols.BumpGrid: (0) Atom type CH3 with ProbeRadius 6 = 1.75\n",
      "protocols.BumpGrid: (0) Atom type aroC with ProbeRadius 6 = 1.75\n",
      "protocols.BumpGrid: (0) Atom type Ntrp with ProbeRadius 4 = 1.55\n",
      "protocols.BumpGrid: (0) Atom type Nhis with ProbeRadius 4 = 1.55\n",
      "protocols.BumpGrid: (0) Atom type NtrR with ProbeRadius 4 = 1.55\n",
      "protocols.BumpGrid: (0) Atom type NH2O with ProbeRadius 4 = 1.55\n",
      "protocols.BumpGrid: (0) Atom type Nlys with ProbeRadius 4 = 1.55\n",
      "protocols.BumpGrid: (0) Atom type Narg with ProbeRadius 4 = 1.55\n",
      "protocols.BumpGrid: (0) Atom type Npro with ProbeRadius 4 = 1.55\n",
      "protocols.BumpGrid: (0) Atom type OH with ProbeRadius 3 = 1.4\n",
      "protocols.BumpGrid: (0) Atom type OW with ProbeRadius 3 = 1.4\n",
      "protocols.BumpGrid: (0) Atom type ONH2 with ProbeRadius 3 = 1.4\n",
      "protocols.BumpGrid: (0) Atom type OOC with ProbeRadius 3 = 1.4\n",
      "protocols.BumpGrid: (0) Atom type Oaro with ProbeRadius 3 = 1.4\n",
      "protocols.BumpGrid: (0) Atom type Oet2 with ProbeRadius 3 = 1.4\n",
      "protocols.BumpGrid: (0) Atom type Oet3 with ProbeRadius 3 = 1.4\n",
      "protocols.BumpGrid: (0) Atom type S with ProbeRadius 7 = 1.85\n",
      "protocols.BumpGrid: (0) Atom type SH1 with ProbeRadius 7 = 1.85\n",
      "protocols.BumpGrid: (0) Atom type Nbb with ProbeRadius 4 = 1.55\n",
      "protocols.BumpGrid: (0) Atom type CAbb with ProbeRadius 6 = 1.75\n",
      "protocols.BumpGrid: (0) Atom type CObb with ProbeRadius 5 = 1.65\n",
      "protocols.BumpGrid: (0) Atom type OCbb with ProbeRadius 3 = 1.4\n",
      "protocols.BumpGrid: (0) Atom type Phos with ProbeRadius 7 = 1.85\n",
      "protocols.BumpGrid: (0) Atom type Pbb with ProbeRadius 7 = 1.85\n",
      "protocols.BumpGrid: (0) Atom type Hpol with ProbeRadius 1 = 1\n",
      "protocols.BumpGrid: (0) Atom type HS with ProbeRadius 1 = 1\n",
      "protocols.BumpGrid: (0) Atom type Hapo with ProbeRadius 2 = 1.17\n",
      "protocols.BumpGrid: (0) Atom type Haro with ProbeRadius 1 = 1\n",
      "protocols.BumpGrid: (0) Atom type HNbb with ProbeRadius 1 = 1\n",
      "protocols.BumpGrid: (0) Atom type Hwat with ProbeRadius 1 = 1\n",
      "protocols.BumpGrid: (0) Atom type Owat with ProbeRadius 3 = 1.4\n",
      "protocols.BumpGrid: (0) Atom type Opoint with ProbeRadius 3 = 1.4\n",
      "protocols.BumpGrid: (0) Atom type HOH with ProbeRadius 3 = 1.4\n",
      "protocols.BumpGrid: (0) Atom type Bsp2 with ProbeRadius 6 = 1.75\n",
      "protocols.BumpGrid: (0) Atom type F with ProbeRadius 3 = 1.4\n",
      "protocols.BumpGrid: (0) Atom type Cl with ProbeRadius 6 = 1.75\n",
      "protocols.BumpGrid: (0) Atom type Br with ProbeRadius 7 = 1.85\n",
      "protocols.BumpGrid: (0) Atom type I with ProbeRadius 7 = 1.85\n",
      "protocols.BumpGrid: (0) Atom type Zn2p with ProbeRadius 0 = 0\n",
      "protocols.BumpGrid: (0) Atom type Co2p with ProbeRadius 2 = 1.17\n",
      "protocols.BumpGrid: (0) Atom type Cu2p with ProbeRadius 0 = 0\n",
      "protocols.BumpGrid: (0) Atom type Fe2p with ProbeRadius 0 = 0\n",
      "protocols.BumpGrid: (0) Atom type Fe3p with ProbeRadius 0 = 0\n",
      "protocols.BumpGrid: (0) Atom type Mg2p with ProbeRadius 1 = 1\n",
      "protocols.BumpGrid: (0) Atom type Ca2p with ProbeRadius 2 = 1.17\n",
      "protocols.BumpGrid: (0) Atom type Pha with ProbeRadius 7 = 1.85\n",
      "protocols.BumpGrid: (0) Atom type OPha with ProbeRadius 3 = 1.4\n",
      "protocols.BumpGrid: (0) Atom type OHha with ProbeRadius 3 = 1.4\n",
      "protocols.BumpGrid: (0) Atom type Hha with ProbeRadius 1 = 1\n",
      "protocols.BumpGrid: (0) Atom type CO3 with ProbeRadius 6 = 1.75\n",
      "protocols.BumpGrid: (0) Atom type OC3 with ProbeRadius 3 = 1.4\n",
      "protocols.BumpGrid: (0) Atom type Si with ProbeRadius 4 = 1.55\n",
      "protocols.BumpGrid: (0) Atom type OSi with ProbeRadius 3 = 1.4\n",
      "protocols.BumpGrid: (0) Atom type Oice with ProbeRadius 3 = 1.4\n",
      "protocols.BumpGrid: (0) Atom type Hice with ProbeRadius 1 = 1\n",
      "protocols.BumpGrid: (0) Atom type Na1p with ProbeRadius 2 = 1.17\n",
      "protocols.BumpGrid: (0) Atom type K1p with ProbeRadius 4 = 1.55\n",
      "protocols.BumpGrid: (0) Atom type He with ProbeRadius 1 = 1\n",
      "protocols.BumpGrid: (0) Atom type Li with ProbeRadius 1 = 1\n",
      "protocols.BumpGrid: (0) Atom type Be with ProbeRadius 2 = 1.17\n",
      "protocols.BumpGrid: (0) Atom type Ne with ProbeRadius 3 = 1.4\n",
      "protocols.BumpGrid: (0) Atom type Al with ProbeRadius 7 = 1.85\n",
      "protocols.BumpGrid: (0) Atom type Ar with ProbeRadius 5 = 1.65\n",
      "protocols.BumpGrid: (0) Atom type Sc with ProbeRadius 3 = 1.4\n",
      "protocols.BumpGrid: (0) Atom type Ti with ProbeRadius 3 = 1.4\n",
      "protocols.BumpGrid: (0) Atom type V with ProbeRadius 2 = 1.17\n",
      "protocols.BumpGrid: (0) Atom type Cr with ProbeRadius 2 = 1.17\n",
      "protocols.BumpGrid: (0) Atom type Mn with ProbeRadius 2 = 1.17\n",
      "protocols.BumpGrid: (0) Atom type Ni with ProbeRadius 2 = 1.17\n",
      "protocols.BumpGrid: (0) Atom type Ga with ProbeRadius 7 = 1.85\n",
      "protocols.BumpGrid: (0) Atom type Ge with ProbeRadius 7 = 1.85\n",
      "protocols.BumpGrid: (0) Atom type As with ProbeRadius 7 = 1.85\n",
      "protocols.BumpGrid: (0) Atom type Se with ProbeRadius 7 = 1.85\n",
      "protocols.BumpGrid: (0) Atom type Kr with ProbeRadius 6 = 1.75\n",
      "protocols.BumpGrid: (0) Atom type Rb with ProbeRadius 6 = 1.75\n",
      "protocols.BumpGrid: (0) Atom type Sr with ProbeRadius 4 = 1.55\n",
      "protocols.BumpGrid: (0) Atom type Y with ProbeRadius 3 = 1.4\n",
      "protocols.BumpGrid: (0) Atom type Zr with ProbeRadius 2 = 1.17\n",
      "protocols.BumpGrid: (0) Atom type Nb with ProbeRadius 3 = 1.4\n",
      "protocols.BumpGrid: (0) Atom type Mo with ProbeRadius 2 = 1.17\n",
      "protocols.BumpGrid: (0) Atom type Tc with ProbeRadius 2 = 1.17\n",
      "protocols.BumpGrid: (0) Atom type Ru with ProbeRadius 2 = 1.17\n",
      "protocols.BumpGrid: (0) Atom type Rh with ProbeRadius 2 = 1.17\n",
      "protocols.BumpGrid: (0) Atom type Pd with ProbeRadius 2 = 1.17\n",
      "protocols.BumpGrid: (0) Atom type Ag with ProbeRadius 3 = 1.4\n",
      "protocols.BumpGrid: (0) Atom type Cd with ProbeRadius 2 = 1.17\n",
      "protocols.BumpGrid: (0) Atom type In with ProbeRadius 7 = 1.85\n",
      "protocols.BumpGrid: (0) Atom type Sn with ProbeRadius 7 = 1.85\n",
      "protocols.BumpGrid: (0) Atom type Sb with ProbeRadius 7 = 1.85\n",
      "protocols.BumpGrid: (0) Atom type Te with ProbeRadius 7 = 1.85\n",
      "protocols.BumpGrid: (0) Atom type Xe with ProbeRadius 7 = 1.85\n",
      "protocols.BumpGrid: (0) Atom type Cs with ProbeRadius 7 = 1.85\n",
      "protocols.BumpGrid: (0) Atom type Ba with ProbeRadius 4 = 1.55\n",
      "protocols.BumpGrid: (0) Atom type La with ProbeRadius 4 = 1.55\n",
      "protocols.BumpGrid: (0) Atom type Ce with ProbeRadius 4 = 1.55\n",
      "protocols.BumpGrid: (0) Atom type Pr with ProbeRadius 4 = 1.55\n",
      "protocols.BumpGrid: (0) Atom type Nd with ProbeRadius 4 = 1.55\n",
      "protocols.BumpGrid: (0) Atom type Pm with ProbeRadius 4 = 1.55\n",
      "protocols.BumpGrid: (0) Atom type Sm with ProbeRadius 4 = 1.55\n",
      "protocols.BumpGrid: (0) Atom type Eu with ProbeRadius 4 = 1.55\n",
      "protocols.BumpGrid: (0) Atom type Gd with ProbeRadius 3 = 1.4\n",
      "protocols.BumpGrid: (0) Atom type Tb with ProbeRadius 3 = 1.4\n",
      "protocols.BumpGrid: (0) Atom type Dy with ProbeRadius 3 = 1.4\n",
      "protocols.BumpGrid: (0) Atom type Ho with ProbeRadius 3 = 1.4\n",
      "protocols.BumpGrid: (0) Atom type Er with ProbeRadius 3 = 1.4\n",
      "protocols.BumpGrid: (0) Atom type Tm with ProbeRadius 3 = 1.4\n",
      "protocols.BumpGrid: (0) Atom type Yb with ProbeRadius 3 = 1.4\n",
      "protocols.BumpGrid: (0) Atom type Lu with ProbeRadius 4 = 1.55\n",
      "protocols.BumpGrid: (0) Atom type Hf with ProbeRadius 2 = 1.17\n",
      "protocols.BumpGrid: (0) Atom type Ta with ProbeRadius 3 = 1.4\n",
      "protocols.BumpGrid: (0) Atom type W with ProbeRadius 2 = 1.17\n",
      "protocols.BumpGrid: (0) Atom type Re with ProbeRadius 2 = 1.17\n",
      "protocols.BumpGrid: (0) Atom type Os with ProbeRadius 2 = 1.17\n",
      "protocols.BumpGrid: (0) Atom type Ir with ProbeRadius 2 = 1.17\n",
      "protocols.BumpGrid: (0) Atom type Pt with ProbeRadius 2 = 1.17\n",
      "protocols.BumpGrid: (0) Atom type Au with ProbeRadius 3 = 1.4\n",
      "protocols.BumpGrid: (0) Atom type Hg with ProbeRadius 2 = 1.17\n",
      "protocols.BumpGrid: (0) Atom type Tl with ProbeRadius 7 = 1.85\n",
      "protocols.BumpGrid: (0) Atom type Pb with ProbeRadius 7 = 1.85\n",
      "protocols.BumpGrid: (0) Atom type Bi with ProbeRadius 7 = 1.85\n",
      "protocols.BumpGrid: (0) Atom type Po with ProbeRadius 7 = 1.85\n",
      "protocols.BumpGrid: (0) Atom type At with ProbeRadius 7 = 1.85\n",
      "protocols.BumpGrid: (0) Atom type Rn with ProbeRadius 7 = 1.85\n",
      "protocols.BumpGrid: (0) Atom type Fr with ProbeRadius 7 = 1.85\n",
      "protocols.BumpGrid: (0) Atom type Ra with ProbeRadius 4 = 1.55\n",
      "protocols.BumpGrid: (0) Atom type Ac with ProbeRadius 3 = 1.4\n",
      "protocols.BumpGrid: (0) Atom type Th with ProbeRadius 3 = 1.4\n",
      "protocols.BumpGrid: (0) Atom type Pa with ProbeRadius 3 = 1.4\n",
      "protocols.BumpGrid: (0) Atom type U with ProbeRadius 3 = 1.4\n",
      "protocols.BumpGrid: (0) Atom type Np with ProbeRadius 3 = 1.4\n",
      "protocols.BumpGrid: (0) Atom type Pu with ProbeRadius 3 = 1.4\n",
      "protocols.BumpGrid: (0) Atom type Am with ProbeRadius 3 = 1.4\n",
      "protocols.BumpGrid: (0) Atom type Cm with ProbeRadius 3 = 1.4\n",
      "protocols.BumpGrid: (0) Atom type Bk with ProbeRadius 3 = 1.4\n",
      "protocols.BumpGrid: (0) Atom type Cf with ProbeRadius 3 = 1.4\n",
      "protocols.BumpGrid: (0) Atom type Es with ProbeRadius 3 = 1.4\n",
      "protocols.BumpGrid: (0) Atom type Fm with ProbeRadius 3 = 1.4\n",
      "protocols.BumpGrid: (0) Atom type Md with ProbeRadius 3 = 1.4\n",
      "protocols.BumpGrid: (0) Atom type No with ProbeRadius 3 = 1.4\n",
      "protocols.BumpGrid: (0) Atom type Lr with ProbeRadius 3 = 1.4\n",
      "protocols.BumpGrid: (0) Atom type SUCK with ProbeRadius 0 = 0\n",
      "protocols.BumpGrid: (0) Atom type REPL with ProbeRadius 2 = 1.17\n",
      "protocols.BumpGrid: (0) Atom type REPLS with ProbeRadius 2 = 1.17\n",
      "protocols.BumpGrid: (0) Atom type HREPS with ProbeRadius 0 = 0\n",
      "protocols.BumpGrid: (0) Atom type VIRT with ProbeRadius 0 = 0\n",
      "protocols.BumpGrid: (0) Atom type MPct with ProbeRadius 0 = 0\n",
      "protocols.BumpGrid: (0) Atom type MPnm with ProbeRadius 0 = 0\n",
      "protocols.BumpGrid: (0) Atom type MPdp with ProbeRadius 0 = 0\n",
      "protocols.BumpGrid: (0) Atom type MPtk with ProbeRadius 0 = 0\n",
      "protocols.match.Matcher: (0) Set occupied space bounding box: Lower (-50.0712, -61.0956, -30.449) Upper (14.845, -6.074, 41.4867)\n",
      "protocols.match.Matcher: (0) Matcher::initialize_from_file: n geometric constraints: 3\n",
      "protocols.match.Matcher: (0) Begin constraint1\n",
      "protocols.match.Matcher: (0)  Upstream residue type GLY for geometric constraint #1\n",
      "protocols.match.Matcher: (0)   Did not locate ALGORITHM_INFO:: match for constraint 1\n",
      "protocols.match.Matcher: (0)    GLY pdb_J1K\n",
      "protocols.match.Matcher: (0)     U3:  C   U2:  CA  U1:  N   D1: O2B  D2: CB   D3: N1B \n",
      "protocols.match.Matcher: (0)      Adding Classical Match Algorithm with geometry samples: \n",
      "protocols.match.Matcher: (0)      tor_U3D1: 246.63 251.63 256.63 261.63 266.63 271.63 276.63\n",
      "protocols.match.Matcher: (0)      ang_U2D1: 103.81 108.81 113.81 118.81 123.81 128.81 133.81\n",
      "protocols.match.Matcher: (0)      dis_U1D1: 2.79\n",
      "protocols.match.Matcher: (0)      tor_U2D2: 301.58 306.58 311.58 316.58 321.58 326.58 331.58\n",
      "protocols.match.Matcher: (0)      ang_U1D2: 85.63 90.63 95.63 100.63 105.63 110.63 115.63\n",
      "protocols.match.Matcher: (0)      tor_U1D3: 253.45 258.45 263.45 268.45 273.45 278.45 283.45\n",
      "protocols.match.Matcher: (0) Begin constraint2\n",
      "protocols.match.Matcher: (0)  Upstream residue type SER for geometric constraint #2\n",
      "protocols.match.Matcher: (0)   Did not locate ALGORITHM_INFO:: match for constraint 2\n",
      "protocols.match.Matcher: (0)    SER pdb_J1K\n",
      "protocols.match.Matcher: (0)     U3:  CA  U2:  CB  U1:  OG  D1: CB   D2: O2B  D3: N1B \n",
      "protocols.match.Matcher: (0)      Adding Classical Match Algorithm with geometry samples: \n",
      "protocols.match.Matcher: (0)      tor_U3D1: 210.28 215.28 220.28 225.28 230.28 235.28 240.28\n",
      "protocols.match.Matcher: (0)      ang_U2D1: 85.14 90.14 95.14 100.14 105.14 110.14 115.14\n",
      "protocols.match.Matcher: (0)      dis_U1D1: 2.75\n",
      "protocols.match.Matcher: (0)      tor_U2D2: 26.74 31.74 36.74 41.74 46.74 51.74 56.74\n",
      "protocols.match.Matcher: (0)      ang_U1D2: 73.52 78.52 83.52 88.52 93.52 98.52 103.52\n",
      "protocols.match.Matcher: (0)      tor_U1D3: 253.87 258.87 263.87 268.87 273.87 278.87 283.87\n",
      "protocols.match.Matcher: (0) Begin constraint3\n",
      "protocols.match.Matcher: (0)  Upstream residue type GLU for geometric constraint #3\n",
      "protocols.match.Matcher: (0)   Did not locate ALGORITHM_INFO:: match for constraint 3\n",
      "protocols.match.Matcher: (0)    GLU pdb_J1K\n",
      "protocols.match.Matcher: (0)     U3:  C   U2:  CA  U1:  N   D1: O2B  D2: CB   D3: N1B \n",
      "protocols.match.Matcher: (0)      Adding Classical Match Algorithm with geometry samples: \n",
      "protocols.match.Matcher: (0)      tor_U3D1: 112.15 117.15 122.15 127.15 132.15 137.15 142.15\n",
      "protocols.match.Matcher: (0)      ang_U2D1: 93.66 98.66 103.66 108.66 113.66 118.66 123.66\n",
      "protocols.match.Matcher: (0)      dis_U1D1: 2.93\n",
      "protocols.match.Matcher: (0)      tor_U2D2: 115.08 120.08 125.08 130.08 135.08 140.08 145.08\n",
      "protocols.match.Matcher: (0)      ang_U1D2: 112.09 117.09 122.09 127.09 132.09 137.09 142.09\n",
      "protocols.match.Matcher: (0)      tor_U1D3: 99.25 104.25 109.25 114.25 119.25 124.25 129.25\n",
      "protocols.match.output.ProcessorFactory: (0) Matches will be consolidated before output.\n",
      "Created UpstreamCollisionFilter\n",
      "Created UpstreamDownstreamCollisionFilter\n",
      "protocols.match.Matcher: (0) Initializing BumpGrids... \n",
      "protocols.match.Matcher: (0) ...done\n",
      "protocols.match.Matcher: (0)  TIMING: Bump grids took 2.80225 seconds to compute\n",
      "core.pack.dunbrack.RotamerLibrary: (0) shapovalov_lib_fixes_enable option is true.\n",
      "core.pack.dunbrack.RotamerLibrary: (0) shapovalov_lib::shap_dun10_smooth_level of 1( aka lowest_smooth ) got activated.\n",
      "core.pack.dunbrack.RotamerLibrary: (0) Binary rotamer library selected: /usr/local/rosetta_mpi/main/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin\n",
      "core.pack.dunbrack.RotamerLibrary: (0) Using Dunbrack library binary file '/usr/local/rosetta_mpi/main/database/rotamer/shapovalov/StpDwn_0-0-0/Dunbrack10.lib.bin'.\n",
      "core.pack.dunbrack.RotamerLibrary: (0) Dunbrack 2010 library took 0.17077 seconds to load from binary\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 1 and found 3833 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 2 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 3 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 4 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 5 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 6 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 7 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 8 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 9 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 10 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 11 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 12 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 13 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 14 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 15 and found 12734 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 16 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 17 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 18 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 19 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 20 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 21 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 22 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 23 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 24 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 25 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 26 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 27 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 28 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 29 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 30 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 31 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 32 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 33 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 34 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 35 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 36 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 37 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 38 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 39 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 40 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 41 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 42 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 43 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 44 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 45 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 46 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 47 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 48 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 49 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 50 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 33614 downstream conformations at residue 51 and found 702 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 52 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 53 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 54 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 55 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 56 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 57 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 33614 downstream conformations at residue 58 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 59 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 60 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 33614 downstream conformations at residue 61 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 62 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 63 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 64 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 65 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 66 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 67 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 68 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 69 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 70 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 71 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 72 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 73 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 74 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 75 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 76 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 77 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 33614 downstream conformations at residue 78 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 79 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 80 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 81 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 82 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 83 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 33614 downstream conformations at residue 84 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 33614 downstream conformations at residue 85 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 86 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 33614 downstream conformations at residue 87 and found 14570 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 88 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 89 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 33614 downstream conformations at residue 90 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 91 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 92 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 93 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 94 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 95 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 96 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 33614 downstream conformations at residue 97 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 98 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 99 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 33614 downstream conformations at residue 100 and found 234 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 33614 downstream conformations at residue 101 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 33614 downstream conformations at residue 102 and found 324 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 103 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 104 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 105 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 106 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 107 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 108 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 109 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 110 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 111 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 112 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 113 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 114 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 115 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 116 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 117 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 118 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 33614 downstream conformations at residue 119 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 33614 downstream conformations at residue 120 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 121 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 122 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 123 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 124 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 125 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 126 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 127 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 128 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 129 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 130 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 131 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 132 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 33614 downstream conformations at residue 133 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 134 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 135 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 136 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 137 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 33614 downstream conformations at residue 138 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 139 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 140 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 141 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 142 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 143 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 144 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 145 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 146 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 147 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 33614 downstream conformations at residue 148 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 149 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 150 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 151 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 152 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 153 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 154 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 155 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 156 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 157 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 158 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 33614 downstream conformations at residue 159 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 160 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 161 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 162 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 163 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 164 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 165 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 166 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 167 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 33614 downstream conformations at residue 168 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 169 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 170 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 171 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 172 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 173 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 174 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 175 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 176 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 177 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 33614 downstream conformations at residue 178 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 179 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 180 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 181 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 33614 downstream conformations at residue 182 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 33614 downstream conformations at residue 183 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 184 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 185 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 33614 downstream conformations at residue 186 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 187 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 188 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 189 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 190 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 191 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 192 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 193 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 194 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 195 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 196 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 197 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 198 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 33614 downstream conformations at residue 199 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 200 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 201 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 202 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 33614 downstream conformations at residue 203 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 204 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 205 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 33614 downstream conformations at residue 206 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 207 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 208 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 209 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 210 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 211 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 212 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 33614 downstream conformations at residue 213 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 214 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 215 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 33614 downstream conformations at residue 216 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 217 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 218 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 219 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 220 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 221 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 222 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 223 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 224 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 225 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 226 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 227 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 228 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 33614 downstream conformations at residue 229 and found 22300 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 230 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 231 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 232 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 233 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 33614 downstream conformations at residue 234 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 235 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 236 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 237 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 238 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 239 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 240 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 241 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 242 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 243 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 244 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 245 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 246 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 247 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 248 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 249 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 250 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 251 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 252 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 253 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 254 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 255 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 33614 downstream conformations at residue 256 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 257 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 258 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 259 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 260 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 261 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 262 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 33614 downstream conformations at residue 263 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 264 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 265 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 266 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 33614 downstream conformations at residue 267 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 268 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 269 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 270 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 271 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 272 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 273 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 274 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 275 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 276 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 277 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 278 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 279 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 280 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 281 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 282 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 283 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 284 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 285 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 286 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 287 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 288 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 289 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 33614 downstream conformations at residue 290 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 291 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 292 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 293 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 294 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 295 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 296 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 33614 downstream conformations at residue 297 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 298 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 299 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 300 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 301 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 302 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 303 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 304 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 305 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 306 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 307 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 308 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 309 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 310 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 311 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 312 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 313 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 314 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 315 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 316 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 317 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 33614 downstream conformations at residue 318 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 319 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 320 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 321 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 322 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 323 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 33614 downstream conformations at residue 324 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 325 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 326 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 327 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 328 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 329 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 330 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 331 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 332 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 333 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 334 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 335 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 33614 downstream conformations at residue 336 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 337 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 338 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 33614 downstream conformations at residue 339 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 340 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 341 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 33614 downstream conformations at residue 342 and found 588 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 343 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 344 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 345 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 346 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 347 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 33614 downstream conformations at residue 348 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 349 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 350 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 351 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 352 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 353 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 354 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 33614 downstream conformations at residue 355 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 356 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 357 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 358 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 359 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 360 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 361 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 362 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 363 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 364 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 365 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 366 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 367 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 368 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 33614 downstream conformations at residue 369 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 370 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 371 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 372 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 373 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 374 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 375 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 376 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 377 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 378 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 379 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 380 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 381 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 382 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 383 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 384 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 385 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 386 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 387 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 388 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 389 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 390 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 391 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 392 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 393 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 394 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 395 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 396 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 397 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 398 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 399 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 400 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 401 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 402 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 403 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 404 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 405 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 406 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 407 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 408 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 409 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 410 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 411 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 33614 downstream conformations at residue 412 and found 22828 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 413 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 414 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 415 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 416 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 417 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 418 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 419 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 420 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 421 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 422 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 33614 downstream conformations at residue 423 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 424 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 425 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 426 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 427 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 428 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 429 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 430 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 431 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 432 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 433 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 434 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 435 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 33614 downstream conformations at residue 436 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 33614 downstream conformations at residue 437 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 438 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 439 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 440 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 441 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 442 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 443 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 33614 downstream conformations at residue 444 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 445 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 446 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 447 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 448 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 449 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 450 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 451 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 452 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 453 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 454 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 455 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 456 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 457 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 458 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 459 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 460 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 461 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 462 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 33614 downstream conformations at residue 463 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 464 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 465 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 466 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 467 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 33614 downstream conformations at residue 468 and found 21232 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 469 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 470 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 33614 downstream conformations at residue 471 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 472 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 473 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 474 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 475 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 476 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 477 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 478 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 479 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 33614 downstream conformations at residue 480 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 481 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 482 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 483 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 484 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 485 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 33614 downstream conformations at residue 486 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 33614 downstream conformations at residue 487 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 488 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 33614 downstream conformations at residue 489 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 490 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 491 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 492 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 493 and found 2 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 494 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 495 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 496 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 497 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 498 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 499 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 500 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 501 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 502 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 503 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 504 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 16807 downstream conformations at residue 505 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamBuilder: (0) Considered 33614 downstream conformations at residue 506 and found 0 hits.\n",
      "protocols.match.upstream.ProteinUpstreamB"
     ]
    },
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "\u001b[0;32m/var/folders/20/jcg_mz1d4czf84ly__lpkd9c0000gn/T/ipykernel_15569/1430948471.py\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m standard_rosetta_match('test_rosetta_match', \n\u001b[0m\u001b[1;32m      2\u001b[0m                        \u001b[0;34m'./test_inputs/6qga_B.pdb'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      3\u001b[0m                        \u001b[0;34m'./test_inputs/1.cst'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      4\u001b[0m                        \u001b[0;34m'pdb_J1K'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      5\u001b[0m                        \u001b[0;34m'./test_inputs/J1K1.params'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/var/folders/20/jcg_mz1d4czf84ly__lpkd9c0000gn/T/ipykernel_15569/4208107040.py\u001b[0m in \u001b[0;36mstandard_rosetta_match\u001b[0;34m(scratching_path, scaffold_pdb, constraint_file, ligand_name, params_file, keep_outputs)\u001b[0m\n\u001b[1;32m    247\u001b[0m         \u001b[0mmatcher_cmd\u001b[0m \u001b[0;34m+=\u001b[0m \u001b[0;34m[\u001b[0m\u001b[0;34m'-extra_res_fa'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mabs_params_file\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    248\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 249\u001b[0;31m     \u001b[0mreturn_code\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0msubprocess\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcall\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mmatcher_cmd\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    250\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    251\u001b[0m     \u001b[0;31m# Count the number of matches and get the matched positions\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/opt/anaconda3/envs/MD/lib/python3.9/subprocess.py\u001b[0m in \u001b[0;36mcall\u001b[0;34m(timeout, *popenargs, **kwargs)\u001b[0m\n\u001b[1;32m    349\u001b[0m     \u001b[0;32mwith\u001b[0m \u001b[0mPopen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0mpopenargs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0mp\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    350\u001b[0m         \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 351\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mwait\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mtimeout\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mtimeout\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    352\u001b[0m         \u001b[0;32mexcept\u001b[0m\u001b[0;34m:\u001b[0m  \u001b[0;31m# Including KeyboardInterrupt, wait handled that.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    353\u001b[0m             \u001b[0mp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mkill\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/opt/anaconda3/envs/MD/lib/python3.9/subprocess.py\u001b[0m in \u001b[0;36mwait\u001b[0;34m(self, timeout)\u001b[0m\n\u001b[1;32m   1187\u001b[0m             \u001b[0mendtime\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0m_time\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;34m+\u001b[0m \u001b[0mtimeout\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1188\u001b[0m         \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1189\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_wait\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mtimeout\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mtimeout\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1190\u001b[0m         \u001b[0;32mexcept\u001b[0m \u001b[0mKeyboardInterrupt\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1191\u001b[0m             \u001b[0;31m# https://bugs.python.org/issue25942\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/opt/anaconda3/envs/MD/lib/python3.9/subprocess.py\u001b[0m in \u001b[0;36m_wait\u001b[0;34m(self, timeout)\u001b[0m\n\u001b[1;32m   1915\u001b[0m                         \u001b[0;32mif\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mreturncode\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1916\u001b[0m                             \u001b[0;32mbreak\u001b[0m  \u001b[0;31m# Another thread waited.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1917\u001b[0;31m                         \u001b[0;34m(\u001b[0m\u001b[0mpid\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0msts\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_try_wait\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1918\u001b[0m                         \u001b[0;31m# Check the pid and loop as waitpid has been known to\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1919\u001b[0m                         \u001b[0;31m# return 0 even without WNOHANG in odd situations.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/opt/anaconda3/envs/MD/lib/python3.9/subprocess.py\u001b[0m in \u001b[0;36m_try_wait\u001b[0;34m(self, wait_flags)\u001b[0m\n\u001b[1;32m   1873\u001b[0m             \u001b[0;34m\"\"\"All callers to this function MUST hold self._waitpid_lock.\"\"\"\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1874\u001b[0m             \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1875\u001b[0;31m                 \u001b[0;34m(\u001b[0m\u001b[0mpid\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0msts\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mos\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mwaitpid\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mpid\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mwait_flags\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1876\u001b[0m             \u001b[0;32mexcept\u001b[0m \u001b[0mChildProcessError\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1877\u001b[0m                 \u001b[0;31m# This happens if SIGCLD is set to be ignored or waiting\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "standard_rosetta_match('test_rosetta_match', \n",
    "                       './test_inputs/6qga_B.pdb',\n",
    "                       './test_inputs/1.cst', \n",
    "                       'pdb_J1K', \n",
    "                       './test_inputs/J1K1.params',\n",
    "                      keep_outputs=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e5049998-e46b-4895-8d05-280f7f70a4ba",
   "metadata": {},
   "source": [
    "## if __name__ == '__main__':\n",
    "    pyrosetta.init()\n",
    "\n",
    "    binding_site_pdb = './test_inputs/test_site.pdb'\n",
    "    site_pose = pyrosetta.pose_from_file(binding_site_pdb) \n",
    "\n",
    "    generate_cst_file(site_pose, './test_inputs/test.cst')\n",
    "    ##write_params_file_for_ligand(binding_site_pdb, 'test.params')\n",
    "\n",
    "    \n",
    "    num_matches, matched_positions = standard_rosetta_match('test_rosetta_match', './test_inputs/2src_scaffold.pdb', './test_inputs/test_short.cst', 'ANP', \n",
    "            './test_inputs/debug_params/params_file_from_mol2/ANP.params',\n",
    "            #keep_outputs=False)\n",
    "            keep_outputs=True)\n",
    "\n",
    "    print('Found {0} matches. The matched positions are {1}'.format(num_matches, matched_positions))"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
