{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "9d47b8d8-edf0-4b16-a5a3-3d37ca6e9c91",
   "metadata": {},
   "outputs": [],
   "source": [
    "#!/usr/bin/env python3\n",
    "'''Match binding sites to scaffolds\n",
    "using standard Rosetta matcher.\n",
    "'''\n",
    "import os\n",
    "import subprocess\n",
    "import shutil\n",
    "import json\n",
    "\n",
    "import numpy as np\n",
    "\n",
    "import pyrosetta\n",
    "from pyrosetta import rosetta\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "48e3b29b-f4ee-47c9-87f8-15c268200353",
   "metadata": {},
   "outputs": [],
   "source": [
    "def angle(x1, x2, x3):\n",
    "    '''Return the angle x1-x2-x3 in degrees.\n",
    "    x1, x2 and x3 are xyzVectors.\n",
    "    '''\n",
    "    v1 = (x1 - x2).normalized()\n",
    "    v2 = (x3 - x2).normalized()\n",
    "\n",
    "    return np.arccos(v1.dot(v2)) * 180 / np.pi\n",
    "\n",
    "def dihedral(x1, x2, x3, x4):\n",
    "    '''Return the dihedral x1-x2-x3-x4 in degrees.\n",
    "    x1, x2, x3 and x4 are xyzVectors.\n",
    "    '''\n",
    "    v1 = (x1 - x2).normalized()\n",
    "    v2 = (x3 - x2).normalized()\n",
    "    v3 = (x4 - x3).normalized()\n",
    "\n",
    "    v1_parallel = v2.normalized()\n",
    "    v1_parallel *= v1.dot(v2)\n",
    "    v3_parallel = v2.normalized()\n",
    "    v3_parallel *= v3.dot(v2)\n",
    "    \n",
    "    v1_n = (v1 - v1_parallel).normalized()\n",
    "    v3_n = (v3 - v3_parallel).normalized()\n",
    "\n",
    "    cos = v1_n.dot(v3_n)\n",
    "    sin = v1_n.cross(v3_n).dot(v2)\n",
    "\n",
    "    return np.arctan2(sin, cos) * 180 / np.pi\n",
    "\n",
    "\n",
    "def atom_residue_distance(pose, res, atom_id, res2):\n",
    "    '''Return the minimum heavy atom distance\n",
    "    between an atom and a residue.\n",
    "    '''\n",
    "    min_distance = float('inf')\n",
    "\n",
    "    for a2 in range(1, pose.residue(res2).nheavyatoms() + 1):\n",
    "        dist = pose.residue(res).xyz(atom_id).distance(pose.residue(res2).xyz(a2))\n",
    "\n",
    "        if dist < min_distance:\n",
    "            min_distance = dist\n",
    "\n",
    "    return min_distance\n",
    "\n",
    "def find_contact_heavy_atoms(pose, res1, res2):\n",
    "    '''Find 3 contact heavy atoms for a residue \n",
    "     i.e. the heavy atoms that is closest to the opposing residue\n",
    "     and the 2 heavy atoms that are closest to \n",
    "     this residue\n",
    "    '''\n",
    "    # Get the atoms that are can be considered as contact atoms for residue 1\n",
    "\n",
    "    potential_contact_atom_ids = []\n",
    "\n",
    "    for a in range(1, pose.residue(res1).nheavyatoms() + 1):\n",
    "        # Exclude protein backbone atoms except for N, CA and C\n",
    "        # If backbon variants are included, for example OXT, the\n",
    "        # matcher application would crash\n",
    "        \n",
    "        if pose.residue(res1).is_protein():\n",
    "            if pose.residue(res1).atom_is_backbone(a):\n",
    "                if pose.residue(res1).atom_name(a).strip() in ['N', 'CA', 'C']:\n",
    "                    potential_contact_atom_ids.append(a)\n",
    "            elif not pose.residue(res1).is_virtual(a): # Ignore virtual atoms for protein residues\n",
    "                potential_contact_atom_ids.append(a)\n",
    "\n",
    "        else:\n",
    "            potential_contact_atom_ids.append(a)\n",
    "\n",
    "    # Calculate distances for all heavy atoms\n",
    "    \n",
    "    distances = []\n",
    "\n",
    "    for a in potential_contact_atom_ids:\n",
    "        distances.append((a, atom_residue_distance(pose, res1, a, res2)))\n",
    "\n",
    "    # Get the names of the closest\n",
    "\n",
    "    distances_sorted = sorted(distances, key=lambda x : x[1])\n",
    "\n",
    "    contact_atoms = []\n",
    "    contact_atoms.append(pose.residue(res1).atom_name(distances_sorted[0][0]))\n",
    "\n",
    "    # Get the two closest intra-residue atoms\n",
    "\n",
    "    intra_distances = []\n",
    "\n",
    "    for a in potential_contact_atom_ids:\n",
    "        if a == distances_sorted[0][0]:\n",
    "            continue\n",
    "\n",
    "        intra_distances.append((a, pose.residue(res1).xyz(distances_sorted[0][0]).distance(\n",
    "            pose.residue(res1).xyz(a))))\n",
    "\n",
    "    intra_distances_sorted = sorted(intra_distances, key=lambda x:x[1])\n",
    "\n",
    "    for i in range(2):\n",
    "        contact_atoms.append(pose.residue(res1).atom_name(intra_distances_sorted[i][0]))\n",
    "\n",
    "    return contact_atoms\n",
    "\n",
    "def generate_cst_file(site_pose, output_file):\n",
    "    '''Generate the constraint file for Rosetta matching.'''\n",
    "    # Find the ligand and protein residues\n",
    "\n",
    "    ligand_residue = 0\n",
    "    protein_residues = []\n",
    "    for i in range(1, site_pose.size() + 1):\n",
    "        if site_pose.residue(i).is_ligand():\n",
    "            ligand_residue = i\n",
    "        elif site_pose.residue(i).is_protein():\n",
    "            protein_residues.append(i)\n",
    "\n",
    "    assert(ligand_residue != 0)\n",
    "\n",
    "    # Generate constraints between the ligand and binding site protein residues \n",
    "\n",
    "    template_sections = \\\n",
    "'''CST::BEGIN\n",
    "  TEMPLATE::   ATOM_MAP: 1 atom_name: {a1_1} {a1_2} {a1_3}\n",
    "  TEMPLATE::   ATOM_MAP: 1 residue3: {r1}\n",
    "\n",
    "  TEMPLATE::   ATOM_MAP: 2 atom_name: {a2_1} {a2_2} {a2_3}\n",
    "  TEMPLATE::   ATOM_MAP: 2 residue3: {r2}\n",
    "\n",
    "  CONSTRAINT:: distanceAB: {dist:7.2f}   0.30 100.00  0        0\n",
    "  CONSTRAINT::    angle_A: {angA:7.2f}  15.00 100.00  360.00   3\n",
    "  CONSTRAINT::    angle_B: {angB:7.2f}  15.00 100.00  360.00   3\n",
    "  CONSTRAINT::  torsion_A: {torA:7.2f}  15.00 100.00  360.00   3\n",
    "  CONSTRAINT::  torsion_B: {torB:7.2f}  15.00 100.00  360.00   3\n",
    "  CONSTRAINT:: torsion_AB: {toAB:7.2f}  15.00 100.00  360.00   3\n",
    "CST::END'''\n",
    "\n",
    "    constraint_sections = []\n",
    "\n",
    "    for pro_res in protein_residues:\n",
    "        ligand_contact_atoms = find_contact_heavy_atoms(site_pose, ligand_residue, pro_res)\n",
    "        protein_contact_atoms = find_contact_heavy_atoms(site_pose, pro_res, ligand_residue)\n",
    "\n",
    "        # Get the positions of contact atoms\n",
    "\n",
    "        v_r1a1 = site_pose.residue(ligand_residue).xyz(ligand_contact_atoms[0])\n",
    "        v_r1a2 = site_pose.residue(ligand_residue).xyz(ligand_contact_atoms[1])\n",
    "        v_r1a3 = site_pose.residue(ligand_residue).xyz(ligand_contact_atoms[2])\n",
    "\n",
    "        v_r2a1 = site_pose.residue(pro_res).xyz(protein_contact_atoms[0])\n",
    "        v_r2a2 = site_pose.residue(pro_res).xyz(protein_contact_atoms[1])\n",
    "        v_r2a3 = site_pose.residue(pro_res).xyz(protein_contact_atoms[2])\n",
    "\n",
    "        # Calculate the constraint distance, angles and torsions\n",
    "\n",
    "        dAB = v_r1a1.distance(v_r2a1)\n",
    "        aA = angle(v_r1a2, v_r1a1, v_r2a1)\n",
    "        aB = angle(v_r1a1, v_r2a1, v_r2a2)\n",
    "        tA = dihedral(v_r1a3, v_r1a2, v_r1a1, v_r2a1)\n",
    "        tB = dihedral(v_r1a1, v_r2a1, v_r2a2, v_r2a3)\n",
    "        tAB = dihedral(v_r1a2, v_r1a1, v_r2a1, v_r2a2)\n",
    "\n",
    "        # Generate the constraint file section\n",
    "\n",
    "        constraint_section = template_sections.format(\n",
    "                a1_1=ligand_contact_atoms[0], a1_2=ligand_contact_atoms[1], a1_3=ligand_contact_atoms[2], r1=site_pose.residue(ligand_residue).name3(),\n",
    "                a2_1=protein_contact_atoms[0], a2_2=protein_contact_atoms[1], a2_3=protein_contact_atoms[2], r2=site_pose.residue(pro_res).name3(),\n",
    "                dist=dAB, angA=aA, angB=aB, torA=tA, torB=tB, toAB=tAB)\n",
    "\n",
    "        constraint_sections.append(constraint_section)\n",
    "\n",
    "    # Write the constraint file\n",
    "\n",
    "    with open(output_file, 'w') as f:\n",
    "        f.write('\\n\\n'.join(constraint_sections))\n",
    "\n",
    "def write_params_file_for_ligand(binding_site_pdb, output_file):\n",
    "    '''Write a params file for the ligand residue in the binding site.\n",
    "    Carefull!!! The default output is not aligned and does not work for matcher.\n",
    "    '''\n",
    "    site_pose = pyrosetta.pose_from_file(binding_site_pdb) \n",
    "    \n",
    "    # Find the ligand and protein residues\n",
    "\n",
    "    ligand_residue = 0\n",
    "    for i in range(1, site_pose.size() + 1):\n",
    "        if site_pose.residue(i).is_ligand():\n",
    "            ligand_residue = i\n",
    "\n",
    "    assert(ligand_residue != 0)\n",
    "\n",
    "    # Write the .params file\n",
    "\n",
    "    rosetta.core.chemical.write_topology_file(site_pose.residue(ligand_residue).type(), output_file)\n",
    "\n",
    "def standard_rosetta_match(scratching_path, scaffold_pdb, constraint_file, ligand_name, params_file=None, keep_outputs=False):\n",
    "    '''Run standard Rosetta matching.\n",
    "    Return the number of matches and the matched positions in rosetta numbering.\n",
    "    Return -1 if matching failed.\n",
    "    '''\n",
    "    abs_scaffold_pdb = os.path.abspath(scaffold_pdb)\n",
    "    abs_constraint_file = os.path.abspath(constraint_file)\n",
    "    \n",
    "    if not (params_file is None):\n",
    "        abs_params_file = os.path.abspath(params_file)\n",
    "   \n",
    "    with open('./site_settings.json', 'r') as f:\n",
    "        site_settings = json.load(f)\n",
    "\n",
    "        matcher_app = site_settings['matcher_app']\n",
    "\n",
    "    # Go to the output path\n",
    "\n",
    "    cwd = os.getcwd()\n",
    "\n",
    "    os.makedirs(scratching_path, exist_ok=True)\n",
    "    os.chdir(scratching_path)\n",
    "\n",
    "    # Generate pos file\n",
    "\n",
    "    scaffold_pose = pyrosetta.pose_from_file(abs_scaffold_pdb)\n",
    "    scaffold_protein_residues = []\n",
    "\n",
    "    for i in range(1, scaffold_pose.size() + 1):\n",
    "        if scaffold_pose.residue(i).is_protein():\n",
    "            scaffold_protein_residues.append(str(i))\n",
    "\n",
    "    pos_file = 'scaffold.pos'\n",
    "    with open(pos_file, 'w') as f:\n",
    "        f.write(' '.join(scaffold_protein_residues))\n",
    "\n",
    "    # Run matcher\n",
    "        \n",
    "    matcher_cmd = [matcher_app,\n",
    "        '-match:output_format', 'PDB',\n",
    "        '-match:match_grouper', 'SameSequenceGrouper',\n",
    "        '-match:consolidate_matches',\n",
    "        '-match:output_matches_per_group', '1',\n",
    "        '-use_input_sc',\n",
    "        '-in:ignore_unrecognized_res',\n",
    "        '-ex1', '-ex2',\n",
    "        '-enumerate_ligand_rotamers', 'false',\n",
    "        '-match::lig_name', ligand_name,\n",
    "        '-match:geometric_constraint_file', abs_constraint_file,\n",
    "        '-s', abs_scaffold_pdb,\n",
    "        '-match::scaffold_active_site_residues', pos_file\n",
    "        ]\n",
    "\n",
    "    if not (params_file is None):\n",
    "        matcher_cmd += ['-extra_res_fa', abs_params_file]\n",
    "\n",
    "    return_code = subprocess.call(matcher_cmd)\n",
    "\n",
    "    # Count the number of matches and get the matched positions\n",
    "\n",
    "    num_matches = 0\n",
    "    matched_positions = []\n",
    "    scaffold_pose = pyrosetta.pose_from_file(abs_scaffold_pdb)\n",
    "    scaffold_pose.pdb_info().set_chains('A')\n",
    "\n",
    "    for f in os.listdir('.'):\n",
    "        if f.startswith('UM') and f.endswith('.pdb'):\n",
    "            num_matches += 1\n",
    "\n",
    "            # Extract the matched positions from the name of the output file\n",
    "\n",
    "            matched_positions.append([])\n",
    "            pos_string = f.split('_')[2]\n",
    "           \n",
    "            # Get the positions of the amino acid characters\n",
    "            c_positions = []\n",
    "            for i in range(len(pos_string)):\n",
    "                if not pos_string[i].isnumeric():\n",
    "                    c_positions.append(i)\n",
    "\n",
    "            for i in range(len(c_positions)):\n",
    "                if i < len(c_positions) - 1:\n",
    "                    pdb_id = int(pos_string[c_positions[i] + 1 : c_positions[i + 1]])\n",
    "                else:\n",
    "                    pdb_id = int(pos_string[c_positions[i] + 1 : len(pos_string)])\n",
    "\n",
    "                matched_positions[-1].append(scaffold_pose.pdb_info().pdb2pose('A', pdb_id))\n",
    "\n",
    "    # Clear the scratch path\n",
    "\n",
    "    os.chdir(cwd)\n",
    "\n",
    "    if not keep_outputs:\n",
    "        shutil.rmtree(scratching_path)\n",
    "\n",
    "    if 0 == return_code:\n",
    "        return num_matches, matched_positions\n",
    "\n",
    "    else:\n",
    "        return -1, matched_positions\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c1cd6dce-05d5-44ea-b22f-2c965b0589c8",
   "metadata": {},
   "source": [
    "### main"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "a3edc994-5c47-44c2-90f7-4d0c88523da6",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyRosetta-4 2021 [Rosetta PyRosetta4.conda.mac.cxx11thread.serialization.python39.Release 2021.38+release.4d5a9693c031951766d575fca9286ed34f560779 2021-09-22T20:45:44] retrieved from: http://www.pyrosetta.org\n",
      "(C) Copyright Rosetta Commons Member Institutions. Created in JHU by Sergey Lyskov and PyRosetta Team.\n",
      "core.init: {0} Checking for fconfig files in pwd and ./rosetta/flags\n",
      "core.init: {0} Rosetta version: PyRosetta4.conda.mac.cxx11thread.serialization.python39.Release r297 2021.38+release.4d5a9693c03 4d5a9693c031951766d575fca9286ed34f560779 http://www.pyrosetta.org 2021-09-22T20:45:44\n",
      "core.init: {0} command: PyRosetta -ex1 -ex2aro -database /Users/chenlin/opt/anaconda3/envs/MD/lib/python3.9/site-packages/pyrosetta/database\n",
      "basic.random.init_random_generator: {0} 'RNG device' seed mode, using '/dev/urandom', seed=1581262691 seed_offset=0 real_seed=1581262691 thread_index=0\n",
      "basic.random.init_random_generator: {0} RandomGenerator:init: Normal mode, seed=1581262691 RG_type=mt19937\n"
     ]
    }
   ],
   "source": [
    "pyrosetta.init()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "e5ac72d7-5390-4a2c-bbc5-de66a53f5491",
   "metadata": {},
   "outputs": [],
   "source": [
    "binding_site_pdb = './test_inputs/3F5J_binding_site.pdb'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "4c29ded5-c873-42c2-a650-bf078fd70485",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.import_pose.import_pose: {0} File './test_inputs/3F5J_binding_site.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: {0} [ WARNING ] missing heavyatom:  OXT on residue GLU:CtermProteinFull 2\n"
     ]
    }
   ],
   "source": [
    "site_pose = pyrosetta.pose_from_file(binding_site_pdb) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "0943afd5-f000-4ed3-ad62-6c53cf7dfea6",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.import_pose.import_pose: {0} File './test_inputs/3F5J_binding_site.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: {0} [ WARNING ] missing heavyatom:  OXT on residue GLU:CtermProteinFull 2\n"
     ]
    }
   ],
   "source": [
    "write_params_file_for_ligand(binding_site_pdb, './test_inputs/BGC.params')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "a9316071-5455-438b-9275-ca7df9a64fcb",
   "metadata": {},
   "outputs": [],
   "source": [
    "generate_cst_file(site_pose, './test_inputs/1.cst')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "29860fdb-4bca-4287-a849-3f7bd7ea54da",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "a3e540ee-9ef7-40c8-9e26-86791888f6df",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "core.import_pose.import_pose: {0} File '/Users/chenlin/OneDrive/thuresearch_git/Match/test_inputs/3F5J_A.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: {0} [ WARNING ] missing heavyatom:  OXT on residue HIS:CtermProteinFull 472\n",
      "core.conformation.Conformation: {0} Found disulfide between residues 191 194\n",
      "core.conformation.Conformation: {0} current variant for 191 CYS\n",
      "core.conformation.Conformation: {0} current variant for 194 CYS\n",
      "core.conformation.Conformation: {0} current variant for 191 CYD\n",
      "core.conformation.Conformation: {0} current variant for 194 CYD\n",
      "core.init: (0) Checking for fconfig files in pwd and ./rosetta/flags \n",
      "core.init: (0) Rosetta version: rosetta.source.release-247 r247 2020.08+release.cb1caba cb1cabafd7463ab703f6abf5efa33d2707b85924 https://www.rosettacommons.org 2020-02-22T18:49:33.865989\n",
      "core.init: (0) command: /usr/local/rosetta_mpi/main/source/bin/match.mpi.macosclangrelease -match:output_format PDB -match:match_grouper SameSequenceGrouper -match:consolidate_matches -match:output_matches_per_group 1 -use_input_sc -in:ignore_unrecognized_res -ex1 -ex2 -enumerate_ligand_rotamers false -match::lig_name pdb_J1K -match:geometric_constraint_file /Users/chenlin/OneDrive/thuresearch_git/Match/test_inputs/1.cst -s /Users/chenlin/OneDrive/thuresearch_git/Match/test_inputs/3F5J_A.pdb -match::scaffold_active_site_residues scaffold.pos -extra_res_fa /Users/chenlin/OneDrive/thuresearch_git/Match/test_inputs/J1K1.params\n",
      "basic.random.init_random_generator: (0) 'RNG device' seed mode, using '/dev/urandom', seed=2144288835 seed_offset=0 real_seed=2144288835\n",
      "basic.random.init_random_generator: (0) RandomGenerator:init: Normal mode, seed=2144288835 RG_type=mt19937\n",
      "core.init: (0) found database environment variable ROSETTA3_DB: /usr/local/rosetta_mpi/main/database\n",
      "core.chemical: (0) [ WARNING ] Ignoring line starting with 'VARIANT_OF' when parsing topology file.\n",
      "core.chemical.GlobalResidueTypeSet: (0) Finished initializing fa_standard residue type set.  Created 981 residue types\n",
      "core.chemical.GlobalResidueTypeSet: (0) Total time to initialize 0.661818 seconds.\n",
      "core.import_pose.import_pose: (0) File '/Users/chenlin/OneDrive/thuresearch_git/Match/test_inputs/3F5J_A.pdb' automatically determined to be of type PDB\n",
      "core.conformation.Conformation: (0) [ WARNING ] missing heavyatom:  OXT on residue HIS:CtermProteinFull 472\n",
      "core.conformation.Conformation: (0) Found disulfide between residues 191 194\n",
      "core.conformation.Conformation: (0) current variant for 191 CYS\n",
      "core.conformation.Conformation: (0) current variant for 194 CYS\n",
      "core.conformation.Conformation: (0) current variant for 191 CYD\n",
      "core.conformation.Conformation: (0) current variant for 194 CYD\n",
      "apps.public.match.match: (0) Selecting orientation atoms: CB   C4   O2B \n",
      "protocols.match.MatcherTask: (0) [ WARNING ] no grid file specified for option -grid_boundary. The bounding grid will be generated from the dimensions of the pose. This is experimental at the moment.\n",
      "protocols.match.MatcherTask: (0) Bounding box lower corner set to (-14.1819,-26.655,11.4925), upper corner set to to (36.106,32.8169,69.4376).\n",
      "core.chemical.GlobalResidueTypeSet: (0) Loading (but possibly not actually using) 'BGC' from the PDB components dictionary for residue type 'pdb_BGC'\n",
      "protocols.toolbox.match_enzdes_util.EnzConstraintIO: (0) read enzyme constraints from /Users/chenlin/OneDrive/thuresearch_git/Match/test_inputs/1.cst ... done, 2 cst blocks were read.\n",
      "protocols.toolbox.match_enzdes_util.AllowedSeqposForGeomCst: (0) Reading match::scaffold_active_stie_residues scaffold.pos: 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40 41 42 43 44 45 46 47 48 49 50 51 52 53 54 55 56 57 58 59 60 61 62 63 64 65 66 67 68 69 70 71 72 73 74 75 76 77 78 79 80 81 82 83 84 85 86 87 88 89 90 91 92 93 94 95 96 97 98 99 100 101 102 103 104 105 106 107 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251 252 253 254 255 256 257 258 259 260 261 262 263 264 265 266 267 268 269 270 271 272 273 274 275 276 277 278 279 280 281 282 283 284 285 286 287 288 289 290 291 292 293 294 295 296 297 298 299 300 301 302 303 304 305 306 307 308 309 310 311 312 313 314 315 316 317 318 319 320 321 322 323 324 325 326 327 328 329 330 331 332 333 334 335 336 337 338 339 340 341 342 343 344 345 346 347 348 349 350 351 352 353 354 355 356 357 358 359 360 361 362 363 364 365 366 367 368 369 370 371 372 373 374 375 376 377 378 379 380 381 382 383 384 385 386 387 388 389 390 391 392 393 394 395 396 397 398 399 400 401 402 403 404 405 406 407 408 409 410 411 412 413 414 415 416 417 418 419 420 421 422 423 424 425 426 427 428 429 430 431 432 433 434 435 436 437 438 439 440 441 442 443 444 445 446 447 448 449 450 451 452 453 454 455 456 457 458 459 460 461 462 463 464 465 466 467 468 469 470 471 472\n",
      "protocols.BumpGrid: (0) Atom type CNH2 with ProbeRadius 5 = 1.65\n",
      "protocols.BumpGrid: (0) Atom type COO with ProbeRadius 5 = 1.65\n",
      "protocols.BumpGrid: (0) Atom type CH0 with ProbeRadius 6 = 1.75\n",
      "protocols.BumpGrid: (0) Atom type CH1 with ProbeRadius 6 = 1.75\n",
      "protocols.BumpGrid: (0) Atom type CH2 with ProbeRadius 6 = 1.75\n",
      "protocols.BumpGrid: (0) Atom type CH3 with ProbeRadius 6 = 1.75\n",
      "protocols.BumpGrid: (0) Atom type aroC with ProbeRadius 6 = 1.75\n",
      "protocols.BumpGrid: (0) Atom type Ntrp with ProbeRadius 4 = 1.55\n",
      "protocols.BumpGrid: (0) Atom type Nhis with ProbeRadius 4 = 1.55\n",
      "protocols.BumpGrid: (0) Atom type NtrR with ProbeRadius 4 = 1.55\n",
      "protocols.BumpGrid: (0) Atom type NH2O with ProbeRadius 4 = 1.55\n",
      "protocols.BumpGrid: (0) Atom type Nlys with ProbeRadius 4 = 1.55\n",
      "protocols.BumpGrid: (0) Atom type Narg with ProbeRadius 4 = 1.55\n",
      "protocols.BumpGrid: (0) Atom type Npro with ProbeRadius 4 = 1.55\n",
      "protocols.BumpGrid: (0) Atom type OH with ProbeRadius 3 = 1.4\n",
      "protocols.BumpGrid: (0) Atom type OW with ProbeRadius 3 = 1.4\n",
      "protocols.BumpGrid: (0) Atom type ONH2 with ProbeRadius 3 = 1.4\n",
      "protocols.BumpGrid: (0) Atom type OOC with ProbeRadius 3 = 1.4\n",
      "protocols.BumpGrid: (0) Atom type Oaro with ProbeRadius 3 = 1.4\n",
      "protocols.BumpGrid: (0) Atom type Oet2 with ProbeRadius 3 = 1.4\n",
      "protocols.BumpGrid: (0) Atom type Oet3 with ProbeRadius 3 = 1.4\n",
      "protocols.BumpGrid: (0) Atom type S with ProbeRadius 7 = 1.85\n",
      "protocols.BumpGrid: (0) Atom type SH1 with ProbeRadius 7 = 1.85\n",
      "protocols.BumpGrid: (0) Atom type Nbb with ProbeRadius 4 = 1.55\n",
      "protocols.BumpGrid: (0) Atom type CAbb with ProbeRadius 6 = 1.75\n",
      "protocols.BumpGrid: (0) Atom type CObb with ProbeRadius 5 = 1.65\n",
      "protocols.BumpGrid: (0) Atom type OCbb with ProbeRadius 3 = 1.4\n",
      "protocols.BumpGrid: (0) Atom type Phos with ProbeRadius 7 = 1.85\n",
      "protocols.BumpGrid: (0) Atom type Pbb with ProbeRadius 7 = 1.85\n",
      "protocols.BumpGrid: (0) Atom type Hpol with ProbeRadius 1 = 1\n",
      "protocols.BumpGrid: (0) Atom type HS with ProbeRadius 1 = 1\n",
      "protocols.BumpGrid: (0) Atom type Hapo with ProbeRadius 2 = 1.17\n",
      "protocols.BumpGrid: (0) Atom type Haro with ProbeRadius 1 = 1\n",
      "protocols.BumpGrid: (0) Atom type HNbb with ProbeRadius 1 = 1\n",
      "protocols.BumpGrid: (0) Atom type Hwat with ProbeRadius 1 = 1\n",
      "protocols.BumpGrid: (0) Atom type Owat with ProbeRadius 3 = 1.4\n",
      "protocols.BumpGrid: (0) Atom type Opoint with ProbeRadius 3 = 1.4\n",
      "protocols.BumpGrid: (0) Atom type HOH with ProbeRadius 3 = 1.4\n",
      "protocols.BumpGrid: (0) Atom type Bsp2 with ProbeRadius 6 = 1.75\n",
      "protocols.BumpGrid: (0) Atom type F with ProbeRadius 3 = 1.4\n",
      "protocols.BumpGrid: (0) Atom type Cl with ProbeRadius 6 = 1.75\n",
      "protocols.BumpGrid: (0) Atom type Br with ProbeRadius 7 = 1.85\n",
      "protocols.BumpGrid: (0) Atom type I with ProbeRadius 7 = 1.85\n",
      "protocols.BumpGrid: (0) Atom type Zn2p with ProbeRadius 0 = 0\n",
      "protocols.BumpGrid: (0) Atom type Co2p with ProbeRadius 2 = 1.17\n",
      "protocols.BumpGrid: (0) Atom type Cu2p with ProbeRadius 0 = 0\n",
      "protocols.BumpGrid: (0) Atom type Fe2p with ProbeRadius 0 = 0\n",
      "protocols.BumpGrid: (0) Atom type Fe3p with ProbeRadius 0 = 0\n",
      "protocols.BumpGrid: (0) Atom type Mg2p with ProbeRadius 1 = 1\n",
      "protocols.BumpGrid: (0) Atom type Ca2p with ProbeRadius 2 = 1.17\n",
      "protocols.BumpGrid: (0) Atom type Pha with ProbeRadius 7 = 1.85\n",
      "protocols.BumpGrid: (0) Atom type OPha with ProbeRadius 3 = 1.4\n",
      "protocols.BumpGrid: (0) Atom type OHha with ProbeRadius 3 = 1.4\n",
      "protocols.BumpGrid: (0) Atom type Hha with ProbeRadius 1 = 1\n",
      "protocols.BumpGrid: (0) Atom type CO3 with ProbeRadius 6 = 1.75\n",
      "protocols.BumpGrid: (0) Atom type OC3 with ProbeRadius 3 = 1.4\n",
      "protocols.BumpGrid: (0) Atom type Si with ProbeRadius 4 = 1.55\n",
      "protocols.BumpGrid: (0) Atom type OSi with ProbeRadius 3 = 1.4\n",
      "protocols.BumpGrid: (0) Atom type Oice with ProbeRadius 3 = 1.4\n",
      "protocols.BumpGrid: (0) Atom type Hice with ProbeRadius 1 = 1\n",
      "protocols.BumpGrid: (0) Atom type Na1p with ProbeRadius 2 = 1.17\n",
      "protocols.BumpGrid: (0) Atom type K1p with ProbeRadius 4 = 1.55\n",
      "protocols.BumpGrid: (0) Atom type He with ProbeRadius 1 = 1\n",
      "protocols.BumpGrid: (0) Atom type Li with ProbeRadius 1 = 1\n",
      "protocols.BumpGrid: (0) Atom type Be with ProbeRadius 2 = 1.17\n",
      "protocols.BumpGrid: (0) Atom type Ne with ProbeRadius 3 = 1.4\n",
      "protocols.BumpGrid: (0) Atom type Al with ProbeRadius 7 = 1.85\n",
      "protocols.BumpGrid: (0) Atom type Ar with ProbeRadius 5 = 1.65\n",
      "protocols.BumpGrid: (0) Atom type Sc with ProbeRadius 3 = 1.4\n",
      "protocols.BumpGrid: (0) Atom type Ti with ProbeRadius 3 = 1.4\n",
      "protocols.BumpGrid: (0) Atom type V with ProbeRadius 2 = 1.17\n",
      "protocols.BumpGrid: (0) Atom type Cr with ProbeRadius 2 = 1.17\n",
      "protocols.BumpGrid: (0) Atom type Mn with ProbeRadius 2 = 1.17\n",
      "protocols.BumpGrid: (0) Atom type Ni with ProbeRadius 2 = 1.17\n",
      "protocols.BumpGrid: (0) Atom type Ga with ProbeRadius 7 = 1.85\n",
      "protocols.BumpGrid: (0) Atom type Ge with ProbeRadius 7 = 1.85\n",
      "protocols.BumpGrid: (0) Atom type As with ProbeRadius 7 = 1.85\n",
      "protocols.BumpGrid: (0) Atom type Se with ProbeRadius 7 = 1.85\n",
      "protocols.BumpGrid: (0) Atom type Kr with ProbeRadius 6 = 1.75\n",
      "protocols.BumpGrid: (0) Atom type Rb with ProbeRadius 6 = 1.75\n",
      "protocols.BumpGrid: (0) Atom type Sr with ProbeRadius 4 = 1.55\n",
      "protocols.BumpGrid: (0) Atom type Y with ProbeRadius 3 = 1.4\n",
      "protocols.BumpGrid: (0) Atom type Zr with ProbeRadius 2 = 1.17\n",
      "protocols.BumpGrid: (0) Atom type Nb with ProbeRadius 3 = 1.4\n",
      "protocols.BumpGrid: (0) Atom type Mo with ProbeRadius 2 = 1.17\n",
      "protocols.BumpGrid: (0) Atom type Tc with ProbeRadius 2 = 1.17\n",
      "protocols.BumpGrid: (0) Atom type Ru with ProbeRadius 2 = 1.17\n",
      "protocols.BumpGrid: (0) Atom type Rh with ProbeRadius 2 = 1.17\n",
      "protocols.BumpGrid: (0) Atom type Pd with ProbeRadius 2 = 1.17\n",
      "protocols.BumpGrid: (0) Atom type Ag with ProbeRadius 3 = 1.4\n",
      "protocols.BumpGrid: (0) Atom type Cd with ProbeRadius 2 = 1.17\n",
      "protocols.BumpGrid: (0) Atom type In with ProbeRadius 7 = 1.85\n",
      "protocols.BumpGrid: (0) Atom type Sn with ProbeRadius 7 = 1.85\n",
      "protocols.BumpGrid: (0) Atom type Sb with ProbeRadius 7 = 1.85\n",
      "protocols.BumpGrid: (0) Atom type Te with ProbeRadius 7 = 1.85\n",
      "protocols.BumpGrid: (0) Atom type Xe with ProbeRadius 7 = 1.85\n",
      "protocols.BumpGrid: (0) Atom type Cs with ProbeRadius 7 = 1.85\n",
      "protocols.BumpGrid: (0) Atom type Ba with ProbeRadius 4 = 1.55\n",
      "protocols.BumpGrid: (0) Atom type La with ProbeRadius 4 = 1.55\n",
      "protocols.BumpGrid: (0) Atom type Ce with ProbeRadius 4 = 1.55\n",
      "protocols.BumpGrid: (0) Atom type Pr with ProbeRadius 4 = 1.55\n",
      "protocols.BumpGrid: (0) Atom type Nd with ProbeRadius 4 = 1.55\n",
      "protocols.BumpGrid: (0) Atom type Pm with ProbeRadius 4 = 1.55\n",
      "protocols.BumpGrid: (0) Atom type Sm with ProbeRadius 4 = 1.55\n",
      "protocols.BumpGrid: (0) Atom type Eu with ProbeRadius 4 = 1.55\n",
      "protocols.BumpGrid: (0) Atom type Gd with ProbeRadius 3 = 1.4\n",
      "protocols.BumpGrid: (0) Atom type Tb with ProbeRadius 3 = 1.4\n",
      "protocols.BumpGrid: (0) Atom type Dy with ProbeRadius 3 = 1.4\n",
      "protocols.BumpGrid: (0) Atom type Ho with ProbeRadius 3 = 1.4\n",
      "protocols.BumpGrid: (0) Atom type Er with ProbeRadius 3 = 1.4\n",
      "protocols.BumpGrid: (0) Atom type Tm with ProbeRadius 3 = 1.4\n",
      "protocols.BumpGrid: (0) Atom type Yb with ProbeRadius 3 = 1.4\n",
      "protocols.BumpGrid: (0) Atom type Lu with ProbeRadius 4 = 1.55\n",
      "protocols.BumpGrid: (0) Atom type Hf with ProbeRadius 2 = 1.17\n",
      "protocols.BumpGrid: (0) Atom type Ta with ProbeRadius 3 = 1.4\n",
      "protocols.BumpGrid: (0) Atom type W with ProbeRadius 2 = 1.17\n",
      "protocols.BumpGrid: (0) Atom type Re with ProbeRadius 2 = 1.17\n",
      "protocols.BumpGrid: (0) Atom type Os with ProbeRadius 2 = 1.17\n",
      "protocols.BumpGrid: (0) Atom type Ir with ProbeRadius 2 = 1.17\n",
      "protocols.BumpGrid: (0) Atom type Pt with ProbeRadius 2 = 1.17\n",
      "protocols.BumpGrid: (0) Atom type Au with ProbeRadius 3 = 1.4\n",
      "protocols.BumpGrid: (0) Atom type Hg with ProbeRadius 2 = 1.17\n",
      "protocols.BumpGrid: (0) Atom type Tl with ProbeRadius 7 = 1.85\n",
      "protocols.BumpGrid: (0) Atom type Pb with ProbeRadius 7 = 1.85\n",
      "protocols.BumpGrid: (0) Atom type Bi with ProbeRadius 7 = 1.85\n",
      "protocols.BumpGrid: (0) Atom type Po with ProbeRadius 7 = 1.85\n",
      "protocols.BumpGrid: (0) Atom type At with ProbeRadius 7 = 1.85\n",
      "protocols.BumpGrid: (0) Atom type Rn with ProbeRadius 7 = 1.85\n",
      "protocols.BumpGrid: (0) Atom type Fr with ProbeRadius 7 = 1.85\n",
      "protocols.BumpGrid: (0) Atom type Ra with ProbeRadius 4 = 1.55\n",
      "protocols.BumpGrid: (0) Atom type Ac with ProbeRadius 3 = 1.4\n",
      "protocols.BumpGrid: (0) Atom type Th with ProbeRadius 3 = 1.4\n",
      "protocols.BumpGrid: (0) Atom type Pa with ProbeRadius 3 = 1.4\n",
      "protocols.BumpGrid: (0) Atom type U with ProbeRadius 3 = 1.4\n",
      "protocols.BumpGrid: (0) Atom type Np with ProbeRadius 3 = 1.4\n",
      "protocols.BumpGrid: (0) Atom type Pu with ProbeRadius 3 = 1.4\n",
      "protocols.BumpGrid: (0) Atom type Am with ProbeRadius 3 = 1.4\n",
      "protocols.BumpGrid: (0) Atom type Cm with ProbeRadius 3 = 1.4\n",
      "protocols.BumpGrid: (0) Atom type Bk with ProbeRadius 3 = 1.4\n",
      "protocols.BumpGrid: (0) Atom type Cf with ProbeRadius 3 = 1.4\n",
      "protocols.BumpGrid: (0) Atom type Es with ProbeRadius 3 = 1.4\n",
      "protocols.BumpGrid: (0) Atom type Fm with ProbeRadius 3 = 1.4\n",
      "protocols.BumpGrid: (0) Atom type Md with ProbeRadius 3 = 1.4\n",
      "protocols.BumpGrid: (0) Atom type No with ProbeRadius 3 = 1.4\n",
      "protocols.BumpGrid: (0) Atom type Lr with ProbeRadius 3 = 1.4\n",
      "protocols.BumpGrid: (0) Atom type SUCK with ProbeRadius 0 = 0\n",
      "protocols.BumpGrid: (0) Atom type REPL with ProbeRadius 2 = 1.17\n",
      "protocols.BumpGrid: (0) Atom type REPLS with ProbeRadius 2 = 1.17\n",
      "protocols.BumpGrid: (0) Atom type HREPS with ProbeRadius 0 = 0\n",
      "protocols.BumpGrid: (0) Atom type VIRT with ProbeRadius 0 = 0\n",
      "protocols.BumpGrid: (0) Atom type MPct with ProbeRadius 0 = 0\n",
      "protocols.BumpGrid: (0) Atom type MPnm with ProbeRadius 0 = 0\n",
      "protocols.BumpGrid: (0) Atom type MPdp with ProbeRadius 0 = 0\n",
      "protocols.BumpGrid: (0) Atom type MPtk with ProbeRadius 0 = 0\n",
      "protocols.match.Matcher: (0) Set occupied space bounding box: Lower (-14.1819, -26.655, 11.4925) Upper (36.106, 32.8169, 69.4376)\n",
      "protocols.match.Matcher: (0) Matcher::initialize_from_file: n geometric constraints: 2\n",
      "protocols.match.Matcher: (0) Begin constraint1\n",
      "protocols.match.Matcher: (0)  Upstream residue type GLN for geometric constraint #1\n",
      "protocols.match.Matcher: (0)   Did not locate ALGORITHM_INFO:: match for constraint 1\n",
      "protocols.match.Matcher: (0)    GLN pdb_BGC\n",
      "protocols.match.Matcher: (0)     U3:  OE1 U2:  CD  U1:  NE2 D1: O4 D2: C4 D3: C3\n",
      "protocols.match.Matcher: (0)      Adding Classical Match Algorithm with geometry samples: \n",
      "protocols.match.Matcher: (0)      tor_U3D1: 282.83 287.83 292.83 297.83 302.83 307.83 312.83\n",
      "protocols.match.Matcher: (0)      ang_U2D1: 87.13 92.13 97.13 102.13 107.13 112.13 117.13\n",
      "protocols.match.Matcher: (0)      dis_U1D1: 11.07\n",
      "protocols.match.Matcher: (0)      tor_U2D2: 178.23 183.23 188.23 193.23 198.23 203.23 208.23\n",
      "protocols.match.Matcher: (0)      ang_U1D2: 120.75 125.75 130.75 135.75 140.75 145.75 150.75\n",
      "protocols.match.Matcher: (0)      tor_U1D3: 76.06 81.06 86.06 91.06 96.06 101.06 106.06\n",
      "protocols.match.Matcher: (0) Begin constraint2\n",
      "protocols.match.Matcher: (0)  Upstream residue type GLU for geometric constraint #2\n",
      "protocols.match.Matcher: (0)   Did not locate ALGORITHM_INFO:: match for constraint 2\n",
      "protocols.match.Matcher: (0)    GLU pdb_BGC\n",
      "protocols.match.Matcher: (0)     U3:  OE2 U2:  CD  U1:  OE1 D1: O4 D2: C4 D3: C3\n",
      "protocols.match.Matcher: (0)      Adding Classical Match Algorithm with geometry samples: \n",
      "protocols.match.Matcher: (0)      tor_U3D1: 61.18 66.18 71.18 76.18 81.18 86.18 91.18\n",
      "protocols.match.Matcher: (0)      ang_U2D1: 129.64 134.64 139.64 144.64 149.64 154.64 159.64\n",
      "protocols.match.Matcher: (0)      dis_U1D1: 14.28\n",
      "protocols.match.Matcher: (0)      tor_U2D2: 210.89 215.89 220.89 225.89 230.89 235.89 240.89\n",
      "protocols.match.Matcher: (0)      ang_U1D2: 127.74 132.74 137.74 142.74 147.74 152.74 157.74\n",
      "protocols.match.Matcher: (0)      tor_U1D3: 102.77 107.77 112.77 117.77 122.77 127.77 132.77\n",
      "protocols.match.output.ProcessorFactory: (0) Matches will be consolidated before output.\n",
      "Created UpstreamCollisionFilter\n",
      "Created UpstreamDownstreamCollisionFilter\n"
     ]
    },
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "\u001b[0;32m/var/folders/20/jcg_mz1d4czf84ly__lpkd9c0000gn/T/ipykernel_15637/1916599750.py\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m standard_rosetta_match('test_rosetta_match', \n\u001b[0m\u001b[1;32m      2\u001b[0m                        \u001b[0;34m'./test_inputs/3F5J_A.pdb'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      3\u001b[0m                        \u001b[0;34m'./test_inputs/1.cst'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      4\u001b[0m                        \u001b[0;34m'pdb_J1K'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      5\u001b[0m                        \u001b[0;34m'./test_inputs/J1K1.params'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/var/folders/20/jcg_mz1d4czf84ly__lpkd9c0000gn/T/ipykernel_15637/4208107040.py\u001b[0m in \u001b[0;36mstandard_rosetta_match\u001b[0;34m(scratching_path, scaffold_pdb, constraint_file, ligand_name, params_file, keep_outputs)\u001b[0m\n\u001b[1;32m    247\u001b[0m         \u001b[0mmatcher_cmd\u001b[0m \u001b[0;34m+=\u001b[0m \u001b[0;34m[\u001b[0m\u001b[0;34m'-extra_res_fa'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mabs_params_file\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    248\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 249\u001b[0;31m     \u001b[0mreturn_code\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0msubprocess\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcall\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mmatcher_cmd\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    250\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    251\u001b[0m     \u001b[0;31m# Count the number of matches and get the matched positions\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/opt/anaconda3/envs/MD/lib/python3.9/subprocess.py\u001b[0m in \u001b[0;36mcall\u001b[0;34m(timeout, *popenargs, **kwargs)\u001b[0m\n\u001b[1;32m    349\u001b[0m     \u001b[0;32mwith\u001b[0m \u001b[0mPopen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0mpopenargs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0mp\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    350\u001b[0m         \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 351\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mwait\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mtimeout\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mtimeout\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    352\u001b[0m         \u001b[0;32mexcept\u001b[0m\u001b[0;34m:\u001b[0m  \u001b[0;31m# Including KeyboardInterrupt, wait handled that.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    353\u001b[0m             \u001b[0mp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mkill\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/opt/anaconda3/envs/MD/lib/python3.9/subprocess.py\u001b[0m in \u001b[0;36mwait\u001b[0;34m(self, timeout)\u001b[0m\n\u001b[1;32m   1187\u001b[0m             \u001b[0mendtime\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0m_time\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;34m+\u001b[0m \u001b[0mtimeout\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1188\u001b[0m         \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1189\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_wait\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mtimeout\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mtimeout\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1190\u001b[0m         \u001b[0;32mexcept\u001b[0m \u001b[0mKeyboardInterrupt\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1191\u001b[0m             \u001b[0;31m# https://bugs.python.org/issue25942\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/opt/anaconda3/envs/MD/lib/python3.9/subprocess.py\u001b[0m in \u001b[0;36m_wait\u001b[0;34m(self, timeout)\u001b[0m\n\u001b[1;32m   1915\u001b[0m                         \u001b[0;32mif\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mreturncode\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1916\u001b[0m                             \u001b[0;32mbreak\u001b[0m  \u001b[0;31m# Another thread waited.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1917\u001b[0;31m                         \u001b[0;34m(\u001b[0m\u001b[0mpid\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0msts\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_try_wait\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1918\u001b[0m                         \u001b[0;31m# Check the pid and loop as waitpid has been known to\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1919\u001b[0m                         \u001b[0;31m# return 0 even without WNOHANG in odd situations.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/opt/anaconda3/envs/MD/lib/python3.9/subprocess.py\u001b[0m in \u001b[0;36m_try_wait\u001b[0;34m(self, wait_flags)\u001b[0m\n\u001b[1;32m   1873\u001b[0m             \u001b[0;34m\"\"\"All callers to this function MUST hold self._waitpid_lock.\"\"\"\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1874\u001b[0m             \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1875\u001b[0;31m                 \u001b[0;34m(\u001b[0m\u001b[0mpid\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0msts\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mos\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mwaitpid\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mpid\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mwait_flags\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1876\u001b[0m             \u001b[0;32mexcept\u001b[0m \u001b[0mChildProcessError\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1877\u001b[0m                 \u001b[0;31m# This happens if SIGCLD is set to be ignored or waiting\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "standard_rosetta_match('test_rosetta_match', \n",
    "                       './test_inputs/3F5J_A.pdb',\n",
    "                       './test_inputs/1.cst', \n",
    "                       'pdb_J1K', \n",
    "                       './test_inputs/J1K1.params',\n",
    "                      keep_outputs=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bc669304-6c59-402a-ad84-45c895fec14e",
   "metadata": {},
   "outputs": [],
   "source": [
    "## if __name__ == '__main__':\n",
    "    pyrosetta.init()\n",
    "\n",
    "    binding_site_pdb = './test_inputs/test_site.pdb'\n",
    "    site_pose = pyrosetta.pose_from_file(binding_site_pdb) \n",
    "\n",
    "    generate_cst_file(site_pose, './test_inputs/test.cst')\n",
    "    ##write_params_file_for_ligand(binding_site_pdb, 'test.params')\n",
    "\n",
    "    \n",
    "    num_matches, matched_positions = standard_rosetta_match('test_rosetta_match', './test_inputs/2src_scaffold.pdb', './test_inputs/test_short.cst', 'ANP', \n",
    "            './test_inputs/debug_params/params_file_from_mol2/ANP.params',\n",
    "            #keep_outputs=False)\n",
    "            keep_outputs=True)\n",
    "\n",
    "    print('Found {0} matches. The matched positions are {1}'.format(num_matches, matched_positions))"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
